From: Kay Sievers <kay@vrfy.org>
Date: Mon, 5 May 2014 14:23:10 +0200
Subject: hwdb: update

(cherry picked from commit 61fb23db45c626d92b4e33f09b9287f58a3625a5)
(cherry picked from commit 65d593c63301cd4deeef72f027901a4511e07f0c)
---
 hwdb/20-OUI.hwdb              | 295 ++++++++++++-
 hwdb/20-pci-vendor-model.hwdb | 174 ++++++--
 hwdb/20-usb-vendor-model.hwdb | 992 ++++++++++++++++++++++++++++++++++++++++--
 3 files changed, 1368 insertions(+), 93 deletions(-)

diff --git a/hwdb/20-OUI.hwdb b/hwdb/20-OUI.hwdb
index 17794ed..afc0d90 100644
--- a/hwdb/20-OUI.hwdb
+++ b/hwdb/20-OUI.hwdb
@@ -13940,7 +13940,7 @@ OUI:000063*
  ID_OUI_FROM_DATABASE=BARCO CONTROL ROOMS GMBH
 
 OUI:000064*
- ID_OUI_FROM_DATABASE=YOKOGAWA DIGITAL COMPUTER CORP
+ ID_OUI_FROM_DATABASE=Yokogawa Electric Corporation
 
 OUI:000065*
  ID_OUI_FROM_DATABASE=Network General Corporation
@@ -14927,7 +14927,7 @@ OUI:0001AE*
  ID_OUI_FROM_DATABASE=Trex Enterprises
 
 OUI:0001AF*
- ID_OUI_FROM_DATABASE=Emerson Network Power
+ ID_OUI_FROM_DATABASE=Artesyn Embedded Technologies
 
 OUI:0001B0*
  ID_OUI_FROM_DATABASE=Fulltek Technology Co., Ltd.
@@ -18272,7 +18272,7 @@ OUI:00060A*
  ID_OUI_FROM_DATABASE=Blue2space
 
 OUI:00060B*
- ID_OUI_FROM_DATABASE=Emerson Network Power
+ ID_OUI_FROM_DATABASE=Artesyn Embedded Technologies
 
 OUI:00060C*
  ID_OUI_FROM_DATABASE=Melco Industries, Inc.
@@ -20441,7 +20441,7 @@ OUI:0008F8*
  ID_OUI_FROM_DATABASE=UTC CCS
 
 OUI:0008F9*
- ID_OUI_FROM_DATABASE=Emerson Network Power
+ ID_OUI_FROM_DATABASE=Artesyn Embedded Technologies
 
 OUI:0008FA*
  ID_OUI_FROM_DATABASE=Karl E.Brinkmann GmbH
@@ -46612,6 +46612,9 @@ OUI:0064A6*
 OUI:00664B*
  ID_OUI_FROM_DATABASE=Huawei Technologies Co., Ltd
 
+OUI:006B8E*
+ ID_OUI_FROM_DATABASE=Shanghai Feixun Communication Co.,Ltd.
+
 OUI:006B9E*
  ID_OUI_FROM_DATABASE=VIZIO Inc
 
@@ -46850,7 +46853,7 @@ OUI:008041*
  ID_OUI_FROM_DATABASE=VEB KOMBINAT ROBOTRON
 
 OUI:008042*
- ID_OUI_FROM_DATABASE=Emerson Network Power
+ ID_OUI_FROM_DATABASE=Artesyn Embedded Technologies
 
 OUI:008043*
  ID_OUI_FROM_DATABASE=NETWORLD, INC.
@@ -49919,7 +49922,7 @@ OUI:00C0F8*
  ID_OUI_FROM_DATABASE=ABOUT COMPUTING INC.
 
 OUI:00C0F9*
- ID_OUI_FROM_DATABASE=Emerson Network Power
+ ID_OUI_FROM_DATABASE=Artesyn Embedded Technologies
 
 OUI:00C0FA*
  ID_OUI_FROM_DATABASE=CANARY COMMUNICATIONS, INC.
@@ -50957,7 +50960,7 @@ OUI:00E034*
  ID_OUI_FROM_DATABASE=CISCO SYSTEMS, INC.
 
 OUI:00E035*
- ID_OUI_FROM_DATABASE=Emerson Network Power
+ ID_OUI_FROM_DATABASE=Artesyn Embedded Technologies
 
 OUI:00E036*
  ID_OUI_FROM_DATABASE=PIONEER CORPORATION
@@ -52477,6 +52480,9 @@ OUI:0838A5*
 OUI:083AB8*
  ID_OUI_FROM_DATABASE=Shinoda Plasma Co., Ltd.
 
+OUI:083D88*
+ ID_OUI_FROM_DATABASE=Samsung Electronics Co.,Ltd
+
 OUI:083E0C*
  ID_OUI_FROM_DATABASE=ARRIS Group, Inc.
 
@@ -52495,9 +52501,6 @@ OUI:084027*
 OUI:08482C*
  ID_OUI_FROM_DATABASE=Raycore Taiwan Co., LTD.
 
-OUI:084929*
- ID_OUI_FROM_DATABASE=CYBATI
-
 OUI:084E1C*
  ID_OUI_FROM_DATABASE=H2A Systems, LLC
 
@@ -52867,6 +52870,9 @@ OUI:0C8910*
 OUI:0C8BFD*
  ID_OUI_FROM_DATABASE=Intel Corporate
 
+OUI:0C8C8F*
+ ID_OUI_FROM_DATABASE=Kamo Technology Limited
+
 OUI:0C8CDC*
  ID_OUI_FROM_DATABASE=Suunto Oy
 
@@ -53515,6 +53521,9 @@ OUI:1499E2*
 OUI:149FE8*
  ID_OUI_FROM_DATABASE=Lenovo Mobile Communication Technology Ltd.
 
+OUI:14A364*
+ ID_OUI_FROM_DATABASE=Samsung Electronics Co.,Ltd
+
 OUI:14A62C*
  ID_OUI_FROM_DATABASE=S.M. Dezac S.A.
 
@@ -54403,6 +54412,9 @@ OUI:20918A*
 OUI:2091D9*
  ID_OUI_FROM_DATABASE=I'M SPA
 
+OUI:20934D*
+ ID_OUI_FROM_DATABASE=Fujian Star-net Communication Co., Ltd
+
 OUI:209AE9*
  ID_OUI_FROM_DATABASE=Volacomm Co., Ltd
 
@@ -54598,6 +54610,9 @@ OUI:24470E*
 OUI:24497B*
  ID_OUI_FROM_DATABASE=Innovative Converged Devices Inc
 
+OUI:244F1D*
+ ID_OUI_FROM_DATABASE=iRule LLC
+
 OUI:245FDF*
  ID_OUI_FROM_DATABASE=KYOCERA Corporation
 
@@ -54751,6 +54766,9 @@ OUI:24DEC6*
 OUI:24E271*
  ID_OUI_FROM_DATABASE=Qingdao Hisense Communications Co.,Ltd
 
+OUI:24E314*
+ ID_OUI_FROM_DATABASE=Apple
+
 OUI:24E6BA*
  ID_OUI_FROM_DATABASE=JSC Zavod im. Kozitsky
 
@@ -54829,6 +54847,9 @@ OUI:2826A6*
 OUI:28285D*
  ID_OUI_FROM_DATABASE=ZyXEL Communications Corporation
 
+OUI:2829CC*
+ ID_OUI_FROM_DATABASE=Corsa Technology Incorporated
+
 OUI:2829D9*
  ID_OUI_FROM_DATABASE=GlobalBeiMing technology (Beijing)Co. Ltd
 
@@ -55087,12 +55108,18 @@ OUI:28E14C*
 OUI:28E297*
  ID_OUI_FROM_DATABASE=Shanghai InfoTM Microelectronics Co.,Ltd.
 
+OUI:28E31F*
+ ID_OUI_FROM_DATABASE=Xiaomi inc.
+
 OUI:28E347*
  ID_OUI_FROM_DATABASE=Liteon Technology Corporation
 
 OUI:28E608*
  ID_OUI_FROM_DATABASE=Tokheim
 
+OUI:28E6E9*
+ ID_OUI_FROM_DATABASE=SIS Sat Internet Services GmbH
+
 OUI:28E794*
  ID_OUI_FROM_DATABASE=Microtime Computer Inc.
 
@@ -55225,6 +55252,9 @@ OUI:2C534A*
 OUI:2C542D*
  ID_OUI_FROM_DATABASE=CISCO SYSTEMS, INC.
 
+OUI:2C54CF*
+ ID_OUI_FROM_DATABASE=LG Electronics
+
 OUI:2C553C*
  ID_OUI_FROM_DATABASE=Gainspeed, Inc.
 
@@ -55330,6 +55360,9 @@ OUI:2CA835*
 OUI:2CAB25*
  ID_OUI_FROM_DATABASE=Shenzhen Gongjin Electronics Co.,Ltd
 
+OUI:2CABA4*
+ ID_OUI_FROM_DATABASE=Cisco SPVTG
+
 OUI:2CB05D*
  ID_OUI_FROM_DATABASE=NETGEAR
 
@@ -55456,6 +55489,9 @@ OUI:303294*
 OUI:3032D4*
  ID_OUI_FROM_DATABASE=Hanilstm Co., Ltd.
 
+OUI:303335*
+ ID_OUI_FROM_DATABASE=Boosty
+
 OUI:3037A6*
  ID_OUI_FROM_DATABASE=CISCO SYSTEMS, INC.
 
@@ -55888,6 +55924,9 @@ OUI:34BA9A*
 OUI:34BB1F*
  ID_OUI_FROM_DATABASE=Research In Motion
 
+OUI:34BB26*
+ ID_OUI_FROM_DATABASE=Motorola Mobility LLC
+
 OUI:34BCA6*
  ID_OUI_FROM_DATABASE=Beijing Ding Qing Technology, Ltd.
 
@@ -55975,6 +56014,9 @@ OUI:34EF44*
 OUI:34EF8B*
  ID_OUI_FROM_DATABASE=NTT Communications Corporation
 
+OUI:34F0CA*
+ ID_OUI_FROM_DATABASE=Shenzhen Linghangyuan Digital Technology Co.,Ltd.
+
 OUI:34F39B*
  ID_OUI_FROM_DATABASE=WizLAN Ltd.
 
@@ -56023,6 +56065,9 @@ OUI:381766*
 OUI:38192F*
  ID_OUI_FROM_DATABASE=Nokia Corporation
 
+OUI:381C1A*
+ ID_OUI_FROM_DATABASE=Cisco
+
 OUI:381C4A*
  ID_OUI_FROM_DATABASE=SIMCom Wireless Solutions Co.,Ltd.
 
@@ -56236,6 +56281,9 @@ OUI:38F597*
 OUI:38F708*
  ID_OUI_FROM_DATABASE=National Resource Management, Inc.
 
+OUI:38F889*
+ ID_OUI_FROM_DATABASE=Huawei Technologies Co., Ltd
+
 OUI:38F8B7*
  ID_OUI_FROM_DATABASE=V2COM PARTICIPACOES S.A.
 
@@ -56287,6 +56335,9 @@ OUI:3C15C2*
 OUI:3C15EA*
  ID_OUI_FROM_DATABASE=TESCOM CO., LTD.
 
+OUI:3C189F*
+ ID_OUI_FROM_DATABASE=Nokia Corporation
+
 OUI:3C18A0*
  ID_OUI_FROM_DATABASE=Luxshare Precision Industry Co.,Ltd.
 
@@ -56521,6 +56572,9 @@ OUI:3CD7DA*
 OUI:3CD92B*
  ID_OUI_FROM_DATABASE=Hewlett-Packard Company
 
+OUI:3CD9CE*
+ ID_OUI_FROM_DATABASE=Eclipse WiFi
+
 OUI:3CDF1E*
  ID_OUI_FROM_DATABASE=CISCO SYSTEMS, INC.
 
@@ -56725,6 +56779,9 @@ OUI:407A80*
 OUI:407B1B*
  ID_OUI_FROM_DATABASE=Mettle Networks Inc.
 
+OUI:408256*
+ ID_OUI_FROM_DATABASE=Continental Automotive GmbH
+
 OUI:4083DE*
  ID_OUI_FROM_DATABASE=Motorola
 
@@ -56818,6 +56875,9 @@ OUI:40C245*
 OUI:40C4D6*
  ID_OUI_FROM_DATABASE=ChongQing Camyu Technology Development Co.,Ltd.
 
+OUI:40C62A*
+ ID_OUI_FROM_DATABASE=Shanghai Jing Ren Electronic Technology Co., Ltd.
+
 OUI:40C7C9*
  ID_OUI_FROM_DATABASE=Naviit Inc.
 
@@ -57001,6 +57061,9 @@ OUI:446132*
 OUI:44619C*
  ID_OUI_FROM_DATABASE=FONsystem co. ltd.
 
+OUI:44666E*
+ ID_OUI_FROM_DATABASE=IP-LINE
+
 OUI:446755*
  ID_OUI_FROM_DATABASE=Orbit Irrigation
 
@@ -57019,6 +57082,9 @@ OUI:44700B*
 OUI:447098*
  ID_OUI_FROM_DATABASE=MING HONG TECHNOLOGY (SHEN ZHEN) LIMITED
 
+OUI:44746C*
+ ID_OUI_FROM_DATABASE=Sony Mobile Communications AB
+
 OUI:447BC4*
  ID_OUI_FROM_DATABASE=DualShine Technology(SZ)Co.,Ltd
 
@@ -57079,6 +57145,9 @@ OUI:44A42D*
 OUI:44A689*
  ID_OUI_FROM_DATABASE=PROMAX ELECTRONICA SA
 
+OUI:44A6E5*
+ ID_OUI_FROM_DATABASE=THINKING TECHNOLOGY CO.,LTD
+
 OUI:44A7CF*
  ID_OUI_FROM_DATABASE=Murata Manufacturing Co., Ltd.
 
@@ -57178,6 +57247,9 @@ OUI:48022A*
 OUI:480362*
  ID_OUI_FROM_DATABASE=DESAY ELECTRONICS(HUIZHOU)CO.,LTD
 
+OUI:480C49*
+ ID_OUI_FROM_DATABASE=NAKAYO TELECOMMUNICATIONS,INC
+
 OUI:481249*
  ID_OUI_FROM_DATABASE=Luxcom Technologies Inc.
 
@@ -57463,6 +57535,9 @@ OUI:4C2258*
 OUI:4C2578*
  ID_OUI_FROM_DATABASE=Nokia Corporation
 
+OUI:4C26E7*
+ ID_OUI_FROM_DATABASE=Welgate Co., Ltd.
+
 OUI:4C2C80*
  ID_OUI_FROM_DATABASE=Beijing Skyway Technologies Co.,Ltd
 
@@ -57535,6 +57610,9 @@ OUI:4C63EB*
 OUI:4C64D9*
  ID_OUI_FROM_DATABASE=Guangdong Leawin Group Co., Ltd
 
+OUI:4C6E6E*
+ ID_OUI_FROM_DATABASE=Comnect Technology CO.,LTD
+
 OUI:4C72B9*
  ID_OUI_FROM_DATABASE=Pegatron Corporation
 
@@ -57565,6 +57643,9 @@ OUI:4C8093*
 OUI:4C82CF*
  ID_OUI_FROM_DATABASE=Echostar Technologies
 
+OUI:4C83DE*
+ ID_OUI_FROM_DATABASE=Cisco SPVTG
+
 OUI:4C8B30*
  ID_OUI_FROM_DATABASE=Actiontec Electronics, Inc
 
@@ -57592,6 +57673,9 @@ OUI:4C9E80*
 OUI:4C9EE4*
  ID_OUI_FROM_DATABASE=Hanyang Navicom Co.,Ltd.
 
+OUI:4C9EFF*
+ ID_OUI_FROM_DATABASE=ZyXEL Communications Corp
+
 OUI:4CA56D*
  ID_OUI_FROM_DATABASE=Samsung Electronics Co.,Ltd
 
@@ -57775,6 +57859,9 @@ OUI:504A6E*
 OUI:504F94*
  ID_OUI_FROM_DATABASE=Loxone Electronics GmbH
 
+OUI:505065*
+ ID_OUI_FROM_DATABASE=TAKT Corporation
+
 OUI:505663*
  ID_OUI_FROM_DATABASE=Texas Instruments
 
@@ -57808,6 +57895,9 @@ OUI:506313*
 OUI:506441*
  ID_OUI_FROM_DATABASE=Greenlee
 
+OUI:506787*
+ ID_OUI_FROM_DATABASE=iTellus
+
 OUI:5067F0*
  ID_OUI_FROM_DATABASE=ZyXEL Communications Corporation
 
@@ -58693,6 +58783,12 @@ OUI:5CAC4C*
 OUI:5CB524*
  ID_OUI_FROM_DATABASE=Sony Ericsson Mobile Communications AB
 
+OUI:5CB6CC*
+ ID_OUI_FROM_DATABASE=NovaComm Technologies Inc.
+
+OUI:5CB8CB*
+ ID_OUI_FROM_DATABASE=Allis Communications
+
 OUI:5CBD9E*
  ID_OUI_FROM_DATABASE=HONGKONG MIRACLE EAGLE TECHNOLOGY(GROUP) LIMITED
 
@@ -58789,6 +58885,9 @@ OUI:5CF8A1*
 OUI:5CF938*
  ID_OUI_FROM_DATABASE=Apple, Inc
 
+OUI:5CF96A*
+ ID_OUI_FROM_DATABASE=Huawei Technologies Co., Ltd
+
 OUI:5CF9DD*
  ID_OUI_FROM_DATABASE=Dell Inc
 
@@ -58927,6 +59026,9 @@ OUI:60748D*
 OUI:607688*
  ID_OUI_FROM_DATABASE=Velodyne
 
+OUI:6077E2*
+ ID_OUI_FROM_DATABASE=Samsung Electronics Co.,Ltd
+
 OUI:60812B*
  ID_OUI_FROM_DATABASE=Custom Control Concepts
 
@@ -59074,6 +59176,9 @@ OUI:60E00E*
 OUI:60E327*
  ID_OUI_FROM_DATABASE=TP-LINK TECHNOLOGIES CO.,LTD.
 
+OUI:60E701*
+ ID_OUI_FROM_DATABASE=Huawei Technologies Co., Ltd
+
 OUI:60E956*
  ID_OUI_FROM_DATABASE=Ayla Networks, Inc
 
@@ -59131,6 +59236,9 @@ OUI:6405BE*
 OUI:64094C*
  ID_OUI_FROM_DATABASE=Beijing Superbee Wireless Technology Co.,Ltd
 
+OUI:640980*
+ ID_OUI_FROM_DATABASE=XIAOMI Electronics,CO.,LTD
+
 OUI:640B4A*
  ID_OUI_FROM_DATABASE=Digital Telecom Technology Limited
 
@@ -59254,6 +59362,9 @@ OUI:645FFF*
 OUI:646223*
  ID_OUI_FROM_DATABASE=Cellient Co., Ltd.
 
+OUI:64649B*
+ ID_OUI_FROM_DATABASE=juniper networks
+
 OUI:6465C0*
  ID_OUI_FROM_DATABASE=Nuvon, Inc
 
@@ -59680,6 +59791,9 @@ OUI:689C5E*
 OUI:689C70*
  ID_OUI_FROM_DATABASE=Apple
 
+OUI:68A0F6*
+ ID_OUI_FROM_DATABASE=Huawei Technologies Co., Ltd
+
 OUI:68A1B7*
  ID_OUI_FROM_DATABASE=Honghao Mingchuan Technology (Beijing) CO.,Ltd.
 
@@ -60130,6 +60244,9 @@ OUI:702B1D*
 OUI:702C1F*
  ID_OUI_FROM_DATABASE=Wisol
 
+OUI:702DD1*
+ ID_OUI_FROM_DATABASE=Newings Communication CO., LTD.
+
 OUI:702F4B*
  ID_OUI_FROM_DATABASE=PolyVision Inc.
 
@@ -60211,6 +60328,9 @@ OUI:705986*
 OUI:705AB6*
  ID_OUI_FROM_DATABASE=COMPAL INFORMATION (KUNSHAN) CO., LTD.
 
+OUI:705B2E*
+ ID_OUI_FROM_DATABASE=M2Communication Inc.
+
 OUI:705CAD*
  ID_OUI_FROM_DATABASE=Konami Gaming Inc
 
@@ -60568,6 +60688,9 @@ OUI:748EF8*
 OUI:748F1B*
  ID_OUI_FROM_DATABASE=MasterImage 3D
 
+OUI:748F4D*
+ ID_OUI_FROM_DATABASE=MEN Mikro Elektronik GmbH
+
 OUI:749050*
  ID_OUI_FROM_DATABASE=Renesas Electronics Corporation
 
@@ -60694,6 +60817,9 @@ OUI:74F07D*
 OUI:74F102*
  ID_OUI_FROM_DATABASE=Beijing HCHCOM Technology Co., Ltd
 
+OUI:74F413*
+ ID_OUI_FROM_DATABASE=Maxwell Forest
+
 OUI:74F612*
  ID_OUI_FROM_DATABASE=ARRIS Group, Inc.
 
@@ -61582,6 +61708,9 @@ OUI:806CBC*
 OUI:80711F*
  ID_OUI_FROM_DATABASE=Juniper Networks
 
+OUI:80717A*
+ ID_OUI_FROM_DATABASE=Huawei Technologies Co., Ltd
+
 OUI:807693*
  ID_OUI_FROM_DATABASE=Newag SA
 
@@ -61633,6 +61762,9 @@ OUI:80946C*
 OUI:8096B1*
  ID_OUI_FROM_DATABASE=ARRIS Group, Inc.
 
+OUI:8096CA*
+ ID_OUI_FROM_DATABASE=Hon Hai Precision Ind Co.,Ltd
+
 OUI:80971B*
  ID_OUI_FROM_DATABASE=Altenergy Power System,Inc.
 
@@ -61699,6 +61831,9 @@ OUI:80D019*
 OUI:80D18B*
  ID_OUI_FROM_DATABASE=Hangzhou I'converge Technology Co.,Ltd
 
+OUI:80D21D*
+ ID_OUI_FROM_DATABASE=AzureWave Technologies, Inc
+
 OUI:80D433*
  ID_OUI_FROM_DATABASE=LzLabs GmbH
 
@@ -62356,6 +62491,9 @@ OUI:8C2F39*
 OUI:8C3330*
  ID_OUI_FROM_DATABASE=EmFirst Co., Ltd.
 
+OUI:8C3357*
+ ID_OUI_FROM_DATABASE=HiteVision Digital Media Technology Co.,Ltd.
+
 OUI:8C3AE3*
  ID_OUI_FROM_DATABASE=LG Electronics
 
@@ -62842,6 +62980,9 @@ OUI:90B134*
 OUI:90B21F*
  ID_OUI_FROM_DATABASE=Apple
 
+OUI:90B686*
+ ID_OUI_FROM_DATABASE=Murata Manufacturing Co., Ltd.
+
 OUI:90B8D0*
  ID_OUI_FROM_DATABASE=Joyent, Inc.
 
@@ -62887,6 +63028,9 @@ OUI:90D92C*
 OUI:90DA4E*
  ID_OUI_FROM_DATABASE=AVANU
 
+OUI:90DA6A*
+ ID_OUI_FROM_DATABASE=MCC System Co., Ltd.
+
 OUI:90DB46*
  ID_OUI_FROM_DATABASE=E-LEAD ELECTRONIC CO., LTD
 
@@ -63100,6 +63244,9 @@ OUI:94ACCA*
 OUI:94AE61*
  ID_OUI_FROM_DATABASE=Alcatel Lucent
 
+OUI:94AEE3*
+ ID_OUI_FROM_DATABASE=Belden Hirschmann Industries (Suzhou) Ltd.
+
 OUI:94B8C5*
  ID_OUI_FROM_DATABASE=RuggedCom Inc.
 
@@ -63151,6 +63298,9 @@ OUI:94CE2C*
 OUI:94D019*
  ID_OUI_FROM_DATABASE=Cydle Corp.
 
+OUI:94D60E*
+ ID_OUI_FROM_DATABASE=shenzhen yunmao information technologies co., ltd
+
 OUI:94D723*
  ID_OUI_FROM_DATABASE=Shanghai DareGlobal Technologies Co., Ltd
 
@@ -63730,6 +63880,9 @@ OUI:9CC077*
 OUI:9CC0D2*
  ID_OUI_FROM_DATABASE=Conductix-Wampfler AG
 
+OUI:9CC172*
+ ID_OUI_FROM_DATABASE=Huawei Technologies Co., Ltd
+
 OUI:9CC7A6*
  ID_OUI_FROM_DATABASE=AVM GmbH
 
@@ -63799,6 +63952,9 @@ OUI:A002DC*
 OUI:A00363*
  ID_OUI_FROM_DATABASE=Robert Bosch Healthcare GmbH
 
+OUI:A00627*
+ ID_OUI_FROM_DATABASE=NEXPA System
+
 OUI:A00798*
  ID_OUI_FROM_DATABASE=Samsung Electronics
 
@@ -63898,6 +64054,9 @@ OUI:A051C6*
 OUI:A055DE*
  ID_OUI_FROM_DATABASE=Pace plc
 
+OUI:A056B2*
+ ID_OUI_FROM_DATABASE=Harman/Becker Automotive Systems GmbH
+
 OUI:A0593A*
  ID_OUI_FROM_DATABASE=V.D.S. Video Display Systems srl
 
@@ -64234,6 +64393,9 @@ OUI:A45C27*
 OUI:A45D36*
  ID_OUI_FROM_DATABASE=Hewlett Packard
 
+OUI:A45DA1*
+ ID_OUI_FROM_DATABASE=ADB Broadband Italia
+
 OUI:A46032*
  ID_OUI_FROM_DATABASE=MRV Communications (Networks) LTD
 
@@ -64312,6 +64474,9 @@ OUI:A49F89*
 OUI:A4A24A*
  ID_OUI_FROM_DATABASE=Cisco SPVTG
 
+OUI:A4A4D3*
+ ID_OUI_FROM_DATABASE=Bluebank Communication Technology Co.Ltd
+
 OUI:A4A80F*
  ID_OUI_FROM_DATABASE=Shenzhen Coship Electronics Co., Ltd.
 
@@ -64462,6 +64627,9 @@ OUI:A80600*
 OUI:A80C0D*
  ID_OUI_FROM_DATABASE=Cisco
 
+OUI:A81374*
+ ID_OUI_FROM_DATABASE=Panasonic Corporation AVC Networks Company
+
 OUI:A8154D*
  ID_OUI_FROM_DATABASE=TP-LINK TECHNOLOGIES CO.,LTD.
 
@@ -64495,6 +64663,9 @@ OUI:A82BD6*
 OUI:A830AD*
  ID_OUI_FROM_DATABASE=Wei Fang Goertek Electronics Co.,Ltd
 
+OUI:A8329A*
+ ID_OUI_FROM_DATABASE=Digicom Futuristic Technologies Ltd.
+
 OUI:A83944*
  ID_OUI_FROM_DATABASE=Actiontec Electronics, Inc
 
@@ -64582,6 +64753,9 @@ OUI:A88CEE*
 OUI:A88D7B*
  ID_OUI_FROM_DATABASE=SunDroid Global limited.
 
+OUI:A88E24*
+ ID_OUI_FROM_DATABASE=Apple
+
 OUI:A8922C*
  ID_OUI_FROM_DATABASE=LG Electronics
 
@@ -64903,6 +65077,9 @@ OUI:AC9CE4*
 OUI:ACA016*
  ID_OUI_FROM_DATABASE=CISCO SYSTEMS, INC.
 
+OUI:ACA213*
+ ID_OUI_FROM_DATABASE=Shenzhen Bilian electronic CO.,LTD
+
 OUI:ACA22C*
  ID_OUI_FROM_DATABASE=Baycity Technologies Ltd
 
@@ -65068,6 +65245,9 @@ OUI:B01B7C*
 OUI:B01C91*
  ID_OUI_FROM_DATABASE=Elim Co
 
+OUI:B01F81*
+ ID_OUI_FROM_DATABASE=IEEE REGISTRATION AUTHORITY  - Please see MAM public listing for more information.
+
 OUI:B024F3*
  ID_OUI_FROM_DATABASE=Progeny Systems
 
@@ -65113,6 +65293,9 @@ OUI:B058C4*
 OUI:B05B1F*
  ID_OUI_FROM_DATABASE=THERMO FISHER SCIENTIFIC S.P.A.
 
+OUI:B05B67*
+ ID_OUI_FROM_DATABASE=Huawei Technologies Co., Ltd
+
 OUI:B05CE5*
  ID_OUI_FROM_DATABASE=Nokia Corporation
 
@@ -65338,6 +65521,9 @@ OUI:B407F9*
 OUI:B40832*
  ID_OUI_FROM_DATABASE=TC Communications
 
+OUI:B40AC6*
+ ID_OUI_FROM_DATABASE=DEXON Systems Ltd.
+
 OUI:B40B44*
  ID_OUI_FROM_DATABASE=Smartisan Technology Co., Ltd.
 
@@ -65533,6 +65719,9 @@ OUI:B4AA4D*
 OUI:B4AB2C*
  ID_OUI_FROM_DATABASE=MtM Technology Corporation
 
+OUI:B4AE6F*
+ ID_OUI_FROM_DATABASE=Circle Reliance, Inc.
+
 OUI:B4B017*
  ID_OUI_FROM_DATABASE=Avaya, Inc
 
@@ -65566,6 +65755,9 @@ OUI:B4C810*
 OUI:B4CCE9*
  ID_OUI_FROM_DATABASE=PROSYST
 
+OUI:B4CEF6*
+ ID_OUI_FROM_DATABASE=HTC Corporation
+
 OUI:B4CFDB*
  ID_OUI_FROM_DATABASE=Shenzhen Jiuzhou Electric Co.,LTD
 
@@ -65812,6 +66004,9 @@ OUI:B898B0*
 OUI:B898F7*
  ID_OUI_FROM_DATABASE=Gionee Communication Equipment Co,Ltd.ShenZhen
 
+OUI:B89919*
+ ID_OUI_FROM_DATABASE=7signal Solutions, Inc
+
 OUI:B89AED*
  ID_OUI_FROM_DATABASE=OceanServer Technology, Inc
 
@@ -66160,6 +66355,9 @@ OUI:BC9889*
 OUI:BC99BC*
  ID_OUI_FROM_DATABASE=FonSee Technology Inc.
 
+OUI:BC9CC5*
+ ID_OUI_FROM_DATABASE=Beijing Huafei Technology Co., Ltd.
+
 OUI:BC9DA5*
  ID_OUI_FROM_DATABASE=DASCOM Europe GmbH
 
@@ -66511,6 +66709,9 @@ OUI:C0EAE4*
 OUI:C0F1C4*
  ID_OUI_FROM_DATABASE=Pacidal Corporation Ltd.
 
+OUI:C0F2FB*
+ ID_OUI_FROM_DATABASE=Apple
+
 OUI:C0F79D*
  ID_OUI_FROM_DATABASE=Powercode
 
@@ -66613,6 +66814,9 @@ OUI:C43C3C*
 OUI:C43DC7*
  ID_OUI_FROM_DATABASE=NETGEAR
 
+OUI:C44202*
+ ID_OUI_FROM_DATABASE=Samsung Electronics Co.,Ltd
+
 OUI:C4438F*
  ID_OUI_FROM_DATABASE=LG Electronics
 
@@ -66634,6 +66838,9 @@ OUI:C44AD0*
 OUI:C44B44*
  ID_OUI_FROM_DATABASE=Omniprint Inc.
 
+OUI:C44BD1*
+ ID_OUI_FROM_DATABASE=Wallys Communications  Teachnologies Co.,Ltd.
+
 OUI:C44E1F*
  ID_OUI_FROM_DATABASE=BlueN
 
@@ -66835,6 +67042,9 @@ OUI:C4F57C*
 OUI:C4FCE4*
  ID_OUI_FROM_DATABASE=DishTV NZ Ltd
 
+OUI:C80210*
+ ID_OUI_FROM_DATABASE=LG Innotek
+
 OUI:C80258*
  ID_OUI_FROM_DATABASE=ITW GSE ApS
 
@@ -67363,6 +67573,9 @@ OUI:CC9F35*
 OUI:CCA0E5*
  ID_OUI_FROM_DATABASE=DZG Metering GmbH
 
+OUI:CCA223*
+ ID_OUI_FROM_DATABASE=Huawei Technologies Co., Ltd
+
 OUI:CCA374*
  ID_OUI_FROM_DATABASE=Guangdong Guanglian Electronic Technology Co.Ltd
 
@@ -67498,6 +67711,9 @@ OUI:CCFE3C*
 OUI:D00790*
  ID_OUI_FROM_DATABASE=Texas Instruments
 
+OUI:D00AAB*
+ ID_OUI_FROM_DATABASE=Yokogawa Digital Computer Corporation
+
 OUI:D00EA4*
  ID_OUI_FROM_DATABASE=Porsche Cars North America
 
@@ -67576,6 +67792,9 @@ OUI:D05875*
 OUI:D059C3*
  ID_OUI_FROM_DATABASE=CeraMicro Technology Corporation
 
+OUI:D059E4*
+ ID_OUI_FROM_DATABASE=Samsung Electronics Co.,Ltd
+
 OUI:D05A0F*
  ID_OUI_FROM_DATABASE=I-BT DIGITAL CO.,LTD
 
@@ -68248,6 +68467,9 @@ OUI:D86595*
 OUI:D866C6*
  ID_OUI_FROM_DATABASE=Shenzhen Daystar Technology Co.,ltd
 
+OUI:D866EE*
+ ID_OUI_FROM_DATABASE=BOXIN COMMUNICATION CO.,LTD.
+
 OUI:D867D9*
  ID_OUI_FROM_DATABASE=CISCO SYSTEMS, INC.
 
@@ -68542,6 +68764,9 @@ OUI:DC49C9*
 OUI:DC4EDE*
  ID_OUI_FROM_DATABASE=SHINYEI TECHNOLOGY CO., LTD.
 
+OUI:DC537C*
+ ID_OUI_FROM_DATABASE=Compal Broadband Networks, Inc.
+
 OUI:DC5726*
  ID_OUI_FROM_DATABASE=Power-One
 
@@ -69112,6 +69337,9 @@ OUI:E45614*
 OUI:E457A8*
  ID_OUI_FROM_DATABASE=Stuart Manufacturing, Inc.
 
+OUI:E45D52*
+ ID_OUI_FROM_DATABASE=Avaya, Inc
+
 OUI:E46449*
  ID_OUI_FROM_DATABASE=ARRIS Group, Inc.
 
@@ -69298,6 +69526,9 @@ OUI:E81132*
 OUI:E81324*
  ID_OUI_FROM_DATABASE=GuangZhou Bonsoninfo System CO.,LTD
 
+OUI:E8150E*
+ ID_OUI_FROM_DATABASE=Nokia Corporation
+
 OUI:E817FC*
  ID_OUI_FROM_DATABASE=NIFTY Corporation
 
@@ -69373,6 +69604,9 @@ OUI:E85B5B*
 OUI:E85BF0*
  ID_OUI_FROM_DATABASE=Imaging Diagnostics
 
+OUI:E85D6B*
+ ID_OUI_FROM_DATABASE=Luminate Wireless
+
 OUI:E85E53*
  ID_OUI_FROM_DATABASE=Infratec Datentechnik GmbH
 
@@ -69409,6 +69643,9 @@ OUI:E878A1*
 OUI:E87AF3*
  ID_OUI_FROM_DATABASE=S5 Tech S.r.l.
 
+OUI:E8802E*
+ ID_OUI_FROM_DATABASE=Apple
+
 OUI:E880D8*
  ID_OUI_FROM_DATABASE=GNTEK Electronics Co.,Ltd.
 
@@ -69719,7 +69956,7 @@ OUI:EC9B5B*
  ID_OUI_FROM_DATABASE=Nokia Corporation
 
 OUI:EC9ECD*
- ID_OUI_FROM_DATABASE=Emerson Network Power and Embedded Computing
+ ID_OUI_FROM_DATABASE=Artesyn Embedded Technologies
 
 OUI:ECA29B*
  ID_OUI_FROM_DATABASE=Kemppi Oy
@@ -69745,6 +69982,9 @@ OUI:ECC38A*
 OUI:ECC882*
  ID_OUI_FROM_DATABASE=CISCO SYSTEMS, INC.
 
+OUI:ECCB30*
+ ID_OUI_FROM_DATABASE=Huawei Technologies Co., Ltd
+
 OUI:ECCD6D*
  ID_OUI_FROM_DATABASE=Allied Telesis, Inc.
 
@@ -69763,6 +70003,9 @@ OUI:ECD925*
 OUI:ECD950*
  ID_OUI_FROM_DATABASE=IRT SA
 
+OUI:ECD9D1*
+ ID_OUI_FROM_DATABASE=Shenzhen TG-NET Botone Technology Co.,Ltd.
+
 OUI:ECDE3D*
  ID_OUI_FROM_DATABASE=Lamprey Networks, Inc.
 
@@ -70144,6 +70387,12 @@ OUI:F406A5*
 OUI:F40B93*
  ID_OUI_FROM_DATABASE=Research In Motion
 
+OUI:F40E11*
+ ID_OUI_FROM_DATABASE=IEEE REGISTRATION AUTHORITY  - Please see MAM public listing for more information.
+
+OUI:F40F1B*
+ ID_OUI_FROM_DATABASE=Cisco
+
 OUI:F40F9B*
  ID_OUI_FROM_DATABASE=WAVELINK
 
@@ -70165,6 +70414,9 @@ OUI:F41FC2*
 OUI:F42012*
  ID_OUI_FROM_DATABASE=Cuciniale GmbH
 
+OUI:F42833*
+ ID_OUI_FROM_DATABASE=MMPC Inc.
+
 OUI:F42896*
  ID_OUI_FROM_DATABASE=SPECTO PAINEIS ELETRONICOS LTDA
 
@@ -70294,6 +70546,9 @@ OUI:F499AC*
 OUI:F49F54*
  ID_OUI_FROM_DATABASE=Samsung Electronics
 
+OUI:F49FF3*
+ ID_OUI_FROM_DATABASE=Huawei Technologies Co., Ltd
+
 OUI:F4A294*
  ID_OUI_FROM_DATABASE=EAGLE WORLD DEVELOPMENT CO., LIMITED
 
@@ -70351,6 +70606,9 @@ OUI:F4CE46*
 OUI:F4CFE2*
  ID_OUI_FROM_DATABASE=Cisco
 
+OUI:F4D032*
+ ID_OUI_FROM_DATABASE=Yunnan Ideal Information&Technology.,Ltd
+
 OUI:F4D261*
  ID_OUI_FROM_DATABASE=SEMOCON Co., Ltd
 
@@ -70390,12 +70648,18 @@ OUI:F4F5A5*
 OUI:F4F5E8*
  ID_OUI_FROM_DATABASE=Google
 
+OUI:F4F646*
+ ID_OUI_FROM_DATABASE=Dediprog Technology Co. Ltd.
+
 OUI:F4F951*
  ID_OUI_FROM_DATABASE=Apple
 
 OUI:F4FC32*
  ID_OUI_FROM_DATABASE=Texas Instruments
 
+OUI:F4FD2B*
+ ID_OUI_FROM_DATABASE=ZOYI Company
+
 OUI:F80113*
  ID_OUI_FROM_DATABASE=Huawei Technologies Co., Ltd
 
@@ -70798,6 +71062,9 @@ OUI:FC1607*
 OUI:FC1794*
  ID_OUI_FROM_DATABASE=InterCreative Co., Ltd
 
+OUI:FC1910*
+ ID_OUI_FROM_DATABASE=Samsung Electronics Co.,Ltd
+
 OUI:FC19D0*
  ID_OUI_FROM_DATABASE=Cloud Vision Networks Technology Co.,Ltd.
 
@@ -70906,6 +71173,9 @@ OUI:FC7516*
 OUI:FC75E6*
  ID_OUI_FROM_DATABASE=Handreamnet
 
+OUI:FC790B*
+ ID_OUI_FROM_DATABASE=Hitachi High Technologies America, Inc.
+
 OUI:FC7CE7*
  ID_OUI_FROM_DATABASE=FCI USA LLC
 
@@ -70948,6 +71218,9 @@ OUI:FCA841*
 OUI:FCA9B0*
  ID_OUI_FROM_DATABASE=MIARTECH (SHANGHAI),INC.
 
+OUI:FCAA14*
+ ID_OUI_FROM_DATABASE=GIGA-BYTE TECHNOLOGY CO.,LTD.
+
 OUI:FCAD0F*
  ID_OUI_FROM_DATABASE=QTS NETWORKS
 
diff --git a/hwdb/20-pci-vendor-model.hwdb b/hwdb/20-pci-vendor-model.hwdb
index 14637d3..2957774 100644
--- a/hwdb/20-pci-vendor-model.hwdb
+++ b/hwdb/20-pci-vendor-model.hwdb
@@ -1673,6 +1673,9 @@ pci:v00001002d00001306*
 pci:v00001002d00001307*
  ID_MODEL_FROM_DATABASE=Kaveri
 
+pci:v00001002d00001308*
+ ID_MODEL_FROM_DATABASE=Kaveri HDMI/DP Audio Controller
+
 pci:v00001002d00001309*
  ID_MODEL_FROM_DATABASE=Kaveri [Radeon R7 Graphics]
 
@@ -4268,6 +4271,9 @@ pci:v00001002d00006611sv00001B0Asd000090D3*
 pci:v00001002d00006613*
  ID_MODEL_FROM_DATABASE=Oland PRO [Radeon R7 240]
 
+pci:v00001002d00006613sv00001682sd00007240*
+ ID_MODEL_FROM_DATABASE=Oland PRO [Radeon R7 240] (R7 240 2048 MB)
+
 pci:v00001002d00006620*
  ID_MODEL_FROM_DATABASE=Mars
 
@@ -6024,31 +6030,31 @@ pci:v00001002d00006819sv0000174Bsd0000E221*
  ID_MODEL_FROM_DATABASE=Pitcairn PRO [Radeon HD 7850] (Radeon HD 7850 2GB GDDR5 DVI-I/DVI-D/HDMI/DP)
 
 pci:v00001002d00006820*
- ID_MODEL_FROM_DATABASE=Venus XTX [Radeon HD 8890M]
+ ID_MODEL_FROM_DATABASE=Venus XTX [Radeon HD 8890M / R9 M275X]
 
 pci:v00001002d00006820sv0000103Csd00001851*
- ID_MODEL_FROM_DATABASE=Venus XTX [Radeon HD 8890M] (Radeon HD 7750M)
+ ID_MODEL_FROM_DATABASE=Venus XTX [Radeon HD 8890M / R9 M275X] (Radeon HD 7750M)
 
 pci:v00001002d00006820sv000017AAsd00003801*
- ID_MODEL_FROM_DATABASE=Venus XTX [Radeon HD 8890M] (Radeon R9 M275)
+ ID_MODEL_FROM_DATABASE=Venus XTX [Radeon HD 8890M / R9 M275X] (Radeon R9 M275)
 
 pci:v00001002d00006821*
- ID_MODEL_FROM_DATABASE=Venus XT [Radeon HD 8870M]
+ ID_MODEL_FROM_DATABASE=Venus XT [Radeon HD 8870M / R9 M270X]
 
 pci:v00001002d00006821sv00001002sd0000031E*
- ID_MODEL_FROM_DATABASE=Venus XT [Radeon HD 8870M] (FirePro SX4000)
+ ID_MODEL_FROM_DATABASE=Venus XT [Radeon HD 8870M / R9 M270X] (FirePro SX4000)
 
 pci:v00001002d00006821sv00001028sd000005CC*
- ID_MODEL_FROM_DATABASE=Venus XT [Radeon HD 8870M] (FirePro M5100)
+ ID_MODEL_FROM_DATABASE=Venus XT [Radeon HD 8870M / R9 M270X] (FirePro M5100)
 
 pci:v00001002d00006821sv00001028sd000015CC*
- ID_MODEL_FROM_DATABASE=Venus XT [Radeon HD 8870M] (FirePro M5100)
+ ID_MODEL_FROM_DATABASE=Venus XT [Radeon HD 8870M / R9 M270X] (FirePro M5100)
 
 pci:v00001002d00006822*
  ID_MODEL_FROM_DATABASE=Venus PRO [Radeon E8860]
 
 pci:v00001002d00006823*
- ID_MODEL_FROM_DATABASE=Venus PRO [Radeon HD 8850M]
+ ID_MODEL_FROM_DATABASE=Venus PRO [Radeon HD 8850M / R9 M265X]
 
 pci:v00001002d00006825*
  ID_MODEL_FROM_DATABASE=Heathrow XT [Radeon HD 7870M]
@@ -7770,64 +7776,73 @@ pci:v00001002d000068F9sv000017AFsd00003014*
  ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 5000/6000/7350/8350 Series] (Radeon HD 6350)
 
 pci:v00001002d000068FA*
- ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350]
+ ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350 / R5 220]
 
 pci:v00001002d000068FAsv00001019sd00000019*
- ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350] (Radeon HD 7350)
+ ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350 / R5 220] (Radeon HD 7350)
 
 pci:v00001002d000068FAsv00001019sd00000021*
- ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350] (Radeon HD 7350)
+ ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350 / R5 220] (Radeon HD 7350)
 
 pci:v00001002d000068FAsv00001019sd00000022*
- ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350] (Radeon HD 7350)
+ ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350 / R5 220] (Radeon HD 7350)
 
 pci:v00001002d000068FAsv00001019sd00000026*
- ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350] (Radeon HD 8350)
+ ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350 / R5 220] (Radeon HD 8350)
 
 pci:v00001002d000068FAsv0000103Csd00002ADF*
- ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350] (Radeon HD 7350A)
+ ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350 / R5 220] (Radeon HD 7350A)
 
 pci:v00001002d000068FAsv0000103Csd00002AE8*
- ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350] (Radeon HD 7350A)
+ ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350 / R5 220] (Radeon HD 7350A)
 
 pci:v00001002d000068FAsv00001043sd00008350*
- ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350] (Radeon HD 8350)
+ ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350 / R5 220] (Radeon HD 8350)
 
 pci:v00001002d000068FAsv00001462sd00002128*
- ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350] (Radeon HD 7350)
+ ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350 / R5 220] (Radeon HD 7350)
 
 pci:v00001002d000068FAsv00001462sd00002184*
- ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350] (Radeon HD 7350)
+ ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350 / R5 220] (Radeon HD 7350)
 
 pci:v00001002d000068FAsv00001462sd00002186*
- ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350] (Radeon HD 7350)
+ ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350 / R5 220] (Radeon HD 7350)
 
 pci:v00001002d000068FAsv00001462sd00002495*
- ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350] (Radeon HD 7350)
+ ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350 / R5 220] (Radeon HD 7350)
 
 pci:v00001002d000068FAsv00001462sd0000B490*
- ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350] (Radeon HD 7350)
+ ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350 / R5 220] (Radeon HD 7350)
 
 pci:v00001002d000068FAsv00001642sd00003985*
- ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350] (Radeon HD 7350)
+ ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350 / R5 220] (Radeon HD 7350)
+
+pci:v00001002d000068FAsv0000174Bsd00003510*
+ ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350 / R5 220] (Radeon HD 8350)
+
+pci:v00001002d000068FAsv0000174Bsd00003521*
+ ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350 / R5 220] (Radeon R5 220)
+
+pci:v00001002d000068FAsv0000174Bsd00003522*
+ ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350 / R5 220] (Radeon R5 220)
 
 pci:v00001002d000068FAsv0000174Bsd00007350*
- ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350] (Radeon HD 7350)
+ ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350 / R5 220] (Radeon HD 7350)
 
 pci:v00001002d000068FAsv0000174Bsd00008153*
- ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350] (Radeon HD 8350)
+ ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350 / R5 220] (Radeon HD 8350)
 
 pci:v00001002d000068FAsv0000174Bsd0000E127*
- ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350] (Radeon HD 7350)
+ ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350 / R5 220] (Radeon HD 7350)
 
 pci:v00001002d000068FAsv0000174Bsd0000E153*
- ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350] (Radeon HD 7350)
+ ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350 / R5 220] (Radeon HD 7350)
 
 pci:v00001002d000068FAsv0000174Bsd0000E180*
- ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350] (Radeon HD 7350)
+ ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350 / R5 220] (Radeon HD 7350)
 
 pci:v00001002d000068FAsv000017AFsd00003015*
- ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350] (Radeon HD 7350)
+ ID_MODEL_FROM_DATABASE=Cedar [Radeon HD 7350/8350 / R5 220] (Radeon HD 7350)
 
 pci:v00001002d000068FE*
  ID_MODEL_FROM_DATABASE=Cedar LE
@@ -8886,7 +8901,7 @@ pci:v00001002d0000980A*
  ID_MODEL_FROM_DATABASE=Wrestler [Radeon HD 7290]
 
 pci:v00001002d00009830*
- ID_MODEL_FROM_DATABASE=Kabini [Radeon HD 8400]
+ ID_MODEL_FROM_DATABASE=Kabini [Radeon HD 8400 / R3 Series]
 
 pci:v00001002d00009831*
  ID_MODEL_FROM_DATABASE=Kabini [Radeon HD 8400E]
@@ -8904,13 +8919,13 @@ pci:v00001002d00009835*
  ID_MODEL_FROM_DATABASE=Kabini [Radeon HD 8310E]
 
 pci:v00001002d00009836*
- ID_MODEL_FROM_DATABASE=Kabini [Radeon HD 8280]
+ ID_MODEL_FROM_DATABASE=Kabini [Radeon HD 8280 / R3 Series]
 
 pci:v00001002d00009837*
  ID_MODEL_FROM_DATABASE=Kabini [Radeon HD 8280E]
 
 pci:v00001002d00009838*
- ID_MODEL_FROM_DATABASE=Kabini [Radeon HD 8240]
+ ID_MODEL_FROM_DATABASE=Kabini [Radeon HD 8240 / R3 Series]
 
 pci:v00001002d00009839*
  ID_MODEL_FROM_DATABASE=Kabini [Radeon HD 8180]
@@ -10724,6 +10739,33 @@ pci:v00001022d00001418*
 pci:v00001022d00001419*
  ID_MODEL_FROM_DATABASE=Family 15h (Models 10h-1fh) I/O Memory Management Unit
 
+pci:v00001022d0000141A*
+ ID_MODEL_FROM_DATABASE=Family 15h (Models 30h-3fh) Processor Function 0
+
+pci:v00001022d0000141B*
+ ID_MODEL_FROM_DATABASE=Family 15h (Models 30h-3fh) Processor Function 1
+
+pci:v00001022d0000141C*
+ ID_MODEL_FROM_DATABASE=Family 15h (Models 30h-3fh) Processor Function 2
+
+pci:v00001022d0000141D*
+ ID_MODEL_FROM_DATABASE=Family 15h (Models 30h-3fh) Processor Function 3
+
+pci:v00001022d0000141E*
+ ID_MODEL_FROM_DATABASE=Family 15h (Models 30h-3fh) Processor Function 4
+
+pci:v00001022d0000141F*
+ ID_MODEL_FROM_DATABASE=Family 15h (Models 30h-3fh) Processor Function 5
+
+pci:v00001022d00001422*
+ ID_MODEL_FROM_DATABASE=Family 15h (Models 30h-3fh) Processor Root Complex
+
+pci:v00001022d00001423*
+ ID_MODEL_FROM_DATABASE=Family 15h (Models 30h-3fh) I/O Memory Management Unit
+
+pci:v00001022d00001426*
+ ID_MODEL_FROM_DATABASE=Family 15h (Models 30h-3fh) Processor Root Port
+
 pci:v00001022d00001439*
  ID_MODEL_FROM_DATABASE=Family 16h Processor Functions 5:1
 
@@ -17621,6 +17663,9 @@ pci:v00001093d000071BC*
 pci:v00001093d000071D0*
  ID_MODEL_FROM_DATABASE=PXI-6143
 
+pci:v00001093d000071DC*
+ ID_MODEL_FROM_DATABASE=PCI-1588
+
 pci:v00001093d00007260*
  ID_MODEL_FROM_DATABASE=PXI-5142
 
@@ -17639,6 +17684,9 @@ pci:v00001093d000072AA*
 pci:v00001093d000072AB*
  ID_MODEL_FROM_DATABASE=PCI-5105
 
+pci:v00001093d000072B8*
+ ID_MODEL_FROM_DATABASE=PXI-6682
+
 pci:v00001093d0000730F*
  ID_MODEL_FROM_DATABASE=PXI-5922EX
 
@@ -17708,6 +17756,9 @@ pci:v00001093d000073F0*
 pci:v00001093d000073F1*
  ID_MODEL_FROM_DATABASE=PCI-5153
 
+pci:v00001093d00007405*
+ ID_MODEL_FROM_DATABASE=PXIe-6674T
+
 pci:v00001093d0000745E*
  ID_MODEL_FROM_DATABASE=PXI-5153EX
 
@@ -17726,6 +17777,12 @@ pci:v00001093d00007539*
 pci:v00001093d0000753A*
  ID_MODEL_FROM_DATABASE=NI 9159
 
+pci:v00001093d000075E5*
+ ID_MODEL_FROM_DATABASE=PXI-6683
+
+pci:v00001093d000075E6*
+ ID_MODEL_FROM_DATABASE=PXI-6683H
+
 pci:v00001093d00007626*
  ID_MODEL_FROM_DATABASE=NI 9154
 
@@ -28100,6 +28157,9 @@ pci:v000010EC*
 pci:v000010ECd00000139*
  ID_MODEL_FROM_DATABASE=RTL-8139/8139C/8139C+ Ethernet Controller
 
+pci:v000010ECd00005208*
+ ID_MODEL_FROM_DATABASE=RTS5208 PCI Express Card Reader
+
 pci:v000010ECd00005209*
  ID_MODEL_FROM_DATABASE=RTS5209 PCI Express Card Reader
 
@@ -28128,7 +28188,7 @@ pci:v000010ECd00005249sv0000103Csd00001909*
  ID_MODEL_FROM_DATABASE=RTS5249 PCI Express Card Reader (ZBook 15)
 
 pci:v000010ECd00005288*
- ID_MODEL_FROM_DATABASE=Barossa PCI Express Card Reader
+ ID_MODEL_FROM_DATABASE=RTS5288 PCI Express Card Reader
 
 pci:v000010ECd00008029*
  ID_MODEL_FROM_DATABASE=RTL-8029(AS)
@@ -32153,6 +32213,9 @@ pci:v00001131d00007160*
 pci:v00001131d00007160sv00001458sd00009009*
  ID_MODEL_FROM_DATABASE=SAA7160 (E8000 DVB-T/Analog TV/FM tuner)
 
+pci:v00001131d00007160sv00001461sd00001455*
+ ID_MODEL_FROM_DATABASE=SAA7160 (AVerTV Hybrid Speedy PCI-E (H788))
+
 pci:v00001131d00007162*
  ID_MODEL_FROM_DATABASE=SAA7162
 
@@ -34691,6 +34754,9 @@ pci:v000011ABd00002A42*
 pci:v000011ABd00002A43*
  ID_MODEL_FROM_DATABASE=88W8366 [TopDog] 802.11n Wireless
 
+pci:v000011ABd00002A55*
+ ID_MODEL_FROM_DATABASE=88W8864 [Avastar] 802.11ac Wireless
+
 pci:v000011ABd00002B36*
  ID_MODEL_FROM_DATABASE=88W8764 [Avastar] 802.11n Wireless
 
@@ -36536,6 +36602,9 @@ pci:v00001217d00008120*
 pci:v00001217d00008130*
  ID_MODEL_FROM_DATABASE=Integrated MS/MSPRO/xD Controller
 
+pci:v00001217d00008221*
+ ID_MODEL_FROM_DATABASE=SD/MMC Card Reader Controller
+
 pci:v00001217d00008320*
  ID_MODEL_FROM_DATABASE=OZ600 MMC/SD Controller
 
@@ -43278,7 +43347,13 @@ pci:v00001469*
  ID_VENDOR_FROM_DATABASE=Cleveland Motion Controls
 
 pci:v0000146A*
- ID_VENDOR_FROM_DATABASE=IFR
+ ID_VENDOR_FROM_DATABASE=Aeroflex
+
+pci:v0000146Ad00003010*
+ ID_MODEL_FROM_DATABASE=3010 RF Synthesizer
+
+pci:v0000146Ad00003A11*
+ ID_MODEL_FROM_DATABASE=3011A PXI RF Synthesizer
 
 pci:v0000146B*
  ID_VENDOR_FROM_DATABASE=Parascan Technologies Ltd
@@ -52409,6 +52484,9 @@ pci:v00001AB8d00004006*
 pci:v00001AB9*
  ID_VENDOR_FROM_DATABASE=Espia Srl
 
+pci:v00001AC8*
+ ID_VENDOR_FROM_DATABASE=Aeroflex Gaisler
+
 pci:v00001ACC*
  ID_VENDOR_FROM_DATABASE=Point of View BV
 
@@ -52724,9 +52802,15 @@ pci:v00001B4Bd000091A0*
 pci:v00001B4Bd000091A4*
  ID_MODEL_FROM_DATABASE=88SE912x IDE Controller
 
+pci:v00001B4Bd00009220*
+ ID_MODEL_FROM_DATABASE=88SE9220 PCIe 2.0 x2 2-port SATA 6 Gb/s RAID Controller
+
 pci:v00001B4Bd00009230*
  ID_MODEL_FROM_DATABASE=88SE9230 PCIe SATA 6Gb/s Controller
 
+pci:v00001B4Bd00009235*
+ ID_MODEL_FROM_DATABASE=88SE9235 PCIe 2.0 x2 4-port SATA 6 Gb/s Controller
+
 pci:v00001B4Bd00009445*
  ID_MODEL_FROM_DATABASE=88SE9445 PCIe 2.0 x4 4-Port SAS/SATA 6 Gbps RAID Controller
 
@@ -54693,7 +54777,7 @@ pci:v00006688*
  ID_VENDOR_FROM_DATABASE=Zycoo Co., Ltd
 
 pci:v00006688d00001200*
- ID_MODEL_FROM_DATABASE=CooVOX TDM Analog Module
+ ID_MODEL_FROM_DATABASE=CooVox TDM Analog Module
 
 pci:v00006688d00001400*
  ID_MODEL_FROM_DATABASE=CooVOX TDM GSM Module
@@ -59066,8 +59150,11 @@ pci:v00008086d00001522sv00008086sd000000A4*
 pci:v00008086d00001523*
  ID_MODEL_FROM_DATABASE=I350 Gigabit Backplane Connection
 
+pci:v00008086d00001523sv00001028sd00000060*
+ ID_MODEL_FROM_DATABASE=I350 Gigabit Backplane Connection (Gigabit 2P I350 LOM)
+
 pci:v00008086d00001523sv00001028sd00001F9B*
- ID_MODEL_FROM_DATABASE=I350 Gigabit Backplane Connection (Gigabit 4P I350 bNDC)
+ ID_MODEL_FROM_DATABASE=I350 Gigabit Backplane Connection (Gigabit 4P I350-t bNDC)
 
 pci:v00008086d00001523sv0000103Csd00001784*
  ID_MODEL_FROM_DATABASE=I350 Gigabit Backplane Connection (Ethernet 1Gb 2-port 361FLB Adapter)
@@ -59183,9 +59270,6 @@ pci:v00008086d00001536*
 pci:v00008086d00001537*
  ID_MODEL_FROM_DATABASE=I210 Gigabit Backplane Connection
 
-pci:v00008086d00001538*
- ID_MODEL_FROM_DATABASE=I210 Gigabit Network Connection
-
 pci:v00008086d00001539*
  ID_MODEL_FROM_DATABASE=I211 Gigabit Network Connection
 
@@ -59270,6 +59354,12 @@ pci:v00008086d0000155Dsv00008086sd00000002*
 pci:v00008086d00001560*
  ID_MODEL_FROM_DATABASE=Ethernet Controller X540
 
+pci:v00008086d0000156F*
+ ID_MODEL_FROM_DATABASE=Ethernet Connection I219-LM
+
+pci:v00008086d00001570*
+ ID_MODEL_FROM_DATABASE=Ethernet Connection I219-V
+
 pci:v00008086d00001571*
  ID_MODEL_FROM_DATABASE=XL710 X710 Virtual Function
 
@@ -59316,16 +59406,16 @@ pci:v00008086d00001581sv00001028sd00001F98*
  ID_MODEL_FROM_DATABASE=Ethernet Controller X710 for 10Gbe backplane (Ethernet 10G 4P X710-k bNDC)
 
 pci:v00008086d00001583*
- ID_MODEL_FROM_DATABASE=Ethernet Controller LX710 for 40GbE QSFP+
+ ID_MODEL_FROM_DATABASE=Ethernet Controller XL710 for 40GbE QSFP+
 
 pci:v00008086d00001583sv00008086sd00000001*
- ID_MODEL_FROM_DATABASE=Ethernet Controller LX710 for 40GbE QSFP+ (Ethernet Converged Network Adapter XL710-Q2)
+ ID_MODEL_FROM_DATABASE=Ethernet Controller XL710 for 40GbE QSFP+ (Ethernet Converged Network Adapter XL710-Q2)
 
 pci:v00008086d00001583sv00008086sd00000002*
- ID_MODEL_FROM_DATABASE=Ethernet Controller LX710 for 40GbE QSFP+ (Ethernet Converged Network Adapter XL710-Q2)
+ ID_MODEL_FROM_DATABASE=Ethernet Controller XL710 for 40GbE QSFP+ (Ethernet Converged Network Adapter XL710-Q2)
 
 pci:v00008086d00001583sv00008086sd00000003*
- ID_MODEL_FROM_DATABASE=Ethernet Controller LX710 for 40GbE QSFP+ (Ethernet I/O Module XL710-Q2)
+ ID_MODEL_FROM_DATABASE=Ethernet Controller XL710 for 40GbE QSFP+ (Ethernet I/O Module XL710-Q2)
 
 pci:v00008086d00001584*
  ID_MODEL_FROM_DATABASE=Ethernet Controller XL710 for 40GbE QSFP+
diff --git a/hwdb/20-usb-vendor-model.hwdb b/hwdb/20-usb-vendor-model.hwdb
index 014c628..52586bc 100644
--- a/hwdb/20-usb-vendor-model.hwdb
+++ b/hwdb/20-usb-vendor-model.hwdb
@@ -21,7 +21,7 @@ usb:v0004*
  ID_VENDOR_FROM_DATABASE=Nebraska Furniture Mart
 
 usb:v0011*
- ID_VENDOR_FROM_DATABASE=Unknown manufacturer
+ ID_VENDOR_FROM_DATABASE=Unknown
 
 usb:v0011p7788*
  ID_MODEL_FROM_DATABASE=Flash mass storage drive
@@ -275,6 +275,9 @@ usb:v03EBp2110*
 usb:v03EBp2122*
  ID_MODEL_FROM_DATABASE=XMEGA-A1 Explained evaluation kit
 
+usb:v03EBp2141*
+ ID_MODEL_FROM_DATABASE=ICE debugger
+
 usb:v03EBp2310*
  ID_MODEL_FROM_DATABASE=EVK11xx evaluation board
 
@@ -455,6 +458,9 @@ usb:v03F0p0121*
 usb:v03F0p0122*
  ID_MODEL_FROM_DATABASE=HID Internet Keyboard
 
+usb:v03F0p0139*
+ ID_MODEL_FROM_DATABASE=Barcode Scanner 4430
+
 usb:v03F0p0201*
  ID_MODEL_FROM_DATABASE=ScanJet 6200c
 
@@ -782,6 +788,9 @@ usb:v03F0p1517*
 usb:v03F0p1524*
  ID_MODEL_FROM_DATABASE=Smart Card Keyboard - KR
 
+usb:v03F0p1539*
+ ID_MODEL_FROM_DATABASE=Mini Magnetic Stripe Reader
+
 usb:v03F0p1602*
  ID_MODEL_FROM_DATABASE=PhotoSmart 330 series
 
@@ -893,6 +902,9 @@ usb:v03F0p1D02*
 usb:v03F0p1D17*
  ID_MODEL_FROM_DATABASE=LaserJet 1320
 
+usb:v03F0p1D24*
+ ID_MODEL_FROM_DATABASE=Barcode scanner
+
 usb:v03F0p1E02*
  ID_MODEL_FROM_DATABASE=PhotoSmart A320 Printer series
 
@@ -935,6 +947,9 @@ usb:v03F0p2012*
 usb:v03F0p201D*
  ID_MODEL_FROM_DATABASE=un2400 Gobi Wireless Modem (QDL mode)
 
+usb:v03F0p2039*
+ ID_MODEL_FROM_DATABASE=Cashdrawer
+
 usb:v03F0p2102*
  ID_MODEL_FROM_DATABASE=PhotoSmart 7345
 
@@ -1008,7 +1023,7 @@ usb:v03F0p2505*
  ID_MODEL_FROM_DATABASE=ScanJet 3770
 
 usb:v03F0p2512*
- ID_MODEL_FROM_DATABASE=OfficeJet Pro L7300
+ ID_MODEL_FROM_DATABASE=OfficeJet Pro L7300 / Compaq LA2405 series monitor
 
 usb:v03F0p2514*
  ID_MODEL_FROM_DATABASE=4-port hub
@@ -1046,6 +1061,9 @@ usb:v03F0p2704*
 usb:v03F0p2717*
  ID_MODEL_FROM_DATABASE=Color LaserJet 2830
 
+usb:v03F0p2724*
+ ID_MODEL_FROM_DATABASE=Magnetic Stripe Reader IDRA-334133-HP
+
 usb:v03F0p2811*
  ID_MODEL_FROM_DATABASE=PSC-2100
 
@@ -1424,6 +1442,9 @@ usb:v03F0p5817*
 usb:v03F0p5911*
  ID_MODEL_FROM_DATABASE=PhotoSmart C6180
 
+usb:v03F0p5912*
+ ID_MODEL_FROM_DATABASE=Officejet Pro 8600
+
 usb:v03F0p5A11*
  ID_MODEL_FROM_DATABASE=PhotoSmart C7100 series
 
@@ -1532,6 +1553,9 @@ usb:v03F0p6B02*
 usb:v03F0p6B11*
  ID_MODEL_FROM_DATABASE=Photosmart C4500 series
 
+usb:v03F0p6C11*
+ ID_MODEL_FROM_DATABASE=Photosmart C4480
+
 usb:v03F0p6C17*
  ID_MODEL_FROM_DATABASE=Color LaserJet 4610
 
@@ -1913,6 +1937,9 @@ usb:v03FC*
 usb:v03FD*
  ID_VENDOR_FROM_DATABASE=Xilinx, Inc.
 
+usb:v03FDp0008*
+ ID_MODEL_FROM_DATABASE=Platform Cable USB II
+
 usb:v03FE*
  ID_VENDOR_FROM_DATABASE=Farallon Comunications
 
@@ -2012,6 +2039,9 @@ usb:v0403p0232*
 usb:v0403p1060*
  ID_MODEL_FROM_DATABASE=JTAG adapter
 
+usb:v0403p1234*
+ ID_MODEL_FROM_DATABASE=IronLogic RFID Adapter [Z-2 USB]
+
 usb:v0403p6001*
  ID_MODEL_FROM_DATABASE=FT232 USB-Serial (UART) IC
 
@@ -2084,12 +2114,18 @@ usb:v0403p8B2B*
 usb:v0403p8B2C*
  ID_MODEL_FROM_DATABASE=Alpermann+Velte TCC70
 
+usb:v0403p9132*
+ ID_MODEL_FROM_DATABASE=LCD and Temperature Interface
+
 usb:v0403p9133*
  ID_MODEL_FROM_DATABASE=CallerID
 
 usb:v0403p9135*
  ID_MODEL_FROM_DATABASE=Rotary Pub alarm
 
+usb:v0403p9136*
+ ID_MODEL_FROM_DATABASE=Pulsecounter
+
 usb:v0403p9E90*
  ID_MODEL_FROM_DATABASE=Marvell OpenRD Base/Client
 
@@ -2291,6 +2327,9 @@ usb:v0403pEF10*
 usb:v0403pF070*
  ID_MODEL_FROM_DATABASE=Serial Converter 422/485 [Vardaan VEUSB422R3]
 
+usb:v0403pF0E9*
+ ID_MODEL_FROM_DATABASE=Tagsys L-P101
+
 usb:v0403pF1A0*
  ID_MODEL_FROM_DATABASE=Asix PRESTO Programmer
 
@@ -2999,6 +3038,9 @@ usb:v040Ap4021*
 usb:v040Ap4022*
  ID_MODEL_FROM_DATABASE=1400 Digital Photo Printer
 
+usb:v040Ap402B*
+ ID_MODEL_FROM_DATABASE=Photo Printer 6850
+
 usb:v040Ap402E*
  ID_MODEL_FROM_DATABASE=605 Photo Printer
 
@@ -4379,6 +4421,9 @@ usb:v0424p7500*
 usb:v0424p9512*
  ID_MODEL_FROM_DATABASE=SMC9512/9514 USB Hub
 
+usb:v0424p9514*
+ ID_MODEL_FROM_DATABASE=SMC9514 Hub
+
 usb:v0424pA700*
  ID_MODEL_FROM_DATABASE=2 Port Hub
 
@@ -5219,6 +5264,9 @@ usb:v0451p2046*
 usb:v0451p2077*
  ID_MODEL_FROM_DATABASE=TUSB2077 Hub
 
+usb:v0451p2F90*
+ ID_MODEL_FROM_DATABASE=SM-USB-DIG
+
 usb:v0451p3410*
  ID_MODEL_FROM_DATABASE=TUSB3410 Microcontroller
 
@@ -5534,6 +5582,9 @@ usb:v0458p5003*
 usb:v0458p5004*
  ID_MODEL_FROM_DATABASE=G-pen Tablet
 
+usb:v0458p505E*
+ ID_MODEL_FROM_DATABASE=Genius iSlim 330
+
 usb:v0458p6001*
  ID_MODEL_FROM_DATABASE=GF3000F Ethernet Adapter
 
@@ -5600,6 +5651,24 @@ usb:v0458p705A*
 usb:v0458p705C*
  ID_MODEL_FROM_DATABASE=Genius iSlim 1300AF
 
+usb:v0458p7061*
+ ID_MODEL_FROM_DATABASE=Genius iLook 1321 V2
+
+usb:v0458p7066*
+ ID_MODEL_FROM_DATABASE=Acer Crystal Eye Webcam
+
+usb:v0458p7067*
+ ID_MODEL_FROM_DATABASE=Genius iSlim 1300AF V2
+
+usb:v0458p7068*
+ ID_MODEL_FROM_DATABASE=Genius eFace 1325R
+
+usb:v0458p706D*
+ ID_MODEL_FROM_DATABASE=Genius iSlim 2000AF V2
+
+usb:v0458p7076*
+ ID_MODEL_FROM_DATABASE=Genius FaceCam 312
+
 usb:v0458p7079*
  ID_MODEL_FROM_DATABASE=FaceCam 2025R
 
@@ -5609,6 +5678,12 @@ usb:v0458p707F*
 usb:v0458p7088*
  ID_MODEL_FROM_DATABASE=WideCam 1050
 
+usb:v0458p7089*
+ ID_MODEL_FROM_DATABASE=Genius FaceCam 320
+
+usb:v0458p708C*
+ ID_MODEL_FROM_DATABASE=Genius WideCam F100
+
 usb:v0459*
  ID_VENDOR_FROM_DATABASE=Adobe Systems, Inc.
 
@@ -6017,6 +6092,15 @@ usb:v045Ep02B0*
 usb:v045Ep02B6*
  ID_MODEL_FROM_DATABASE=Xbox 360 / Bluetooth Wireless Headset
 
+usb:v045Ep02BE*
+ ID_MODEL_FROM_DATABASE=Kinect for Windows NUI Audio
+
+usb:v045Ep02BF*
+ ID_MODEL_FROM_DATABASE=Kinect for Windows NUI Camera
+
+usb:v045Ep02C2*
+ ID_MODEL_FROM_DATABASE=Kinect for Windows NUI Motor
+
 usb:v045Ep0400*
  ID_MODEL_FROM_DATABASE=Windows Powered Pocket PC 2002
 
@@ -6866,6 +6950,9 @@ usb:v046Dp0830*
 usb:v046Dp0840*
  ID_MODEL_FROM_DATABASE=QuickCam Express
 
+usb:v046Dp0843*
+ ID_MODEL_FROM_DATABASE=Webcam C930e
+
 usb:v046Dp0850*
  ID_MODEL_FROM_DATABASE=QuickCam Web
 
@@ -7169,6 +7256,12 @@ usb:v046Dp0A1F*
 usb:v046Dp0A29*
  ID_MODEL_FROM_DATABASE=H600 [Wireless Headset]
 
+usb:v046Dp0A38*
+ ID_MODEL_FROM_DATABASE=Headset H340
+
+usb:v046Dp0A4D*
+ ID_MODEL_FROM_DATABASE=G430 Surround Sound Gaming Headset
+
 usb:v046Dp0B02*
  ID_MODEL_FROM_DATABASE=C-UV35 [Bluetooth Mini-Receiver] (HID proxy mode)
 
@@ -7650,7 +7743,7 @@ usb:v046DpC31B*
  ID_MODEL_FROM_DATABASE=Compact Keyboard K300
 
 usb:v046DpC31C*
- ID_MODEL_FROM_DATABASE=Keyboard K120 for Business
+ ID_MODEL_FROM_DATABASE=Keyboard K120
 
 usb:v046DpC31D*
  ID_MODEL_FROM_DATABASE=Media Keyboard K200
@@ -7751,6 +7844,9 @@ usb:v046DpC529*
 usb:v046DpC52B*
  ID_MODEL_FROM_DATABASE=Unifying Receiver
 
+usb:v046DpC52D*
+ ID_MODEL_FROM_DATABASE=R700 Remote Presenter receiver
+
 usb:v046DpC52E*
  ID_MODEL_FROM_DATABASE=MK260 Wireless Combo Receiver
 
@@ -7760,6 +7856,18 @@ usb:v046DpC52F*
 usb:v046DpC532*
  ID_MODEL_FROM_DATABASE=Unifying Receiver
 
+usb:v046DpC603*
+ ID_MODEL_FROM_DATABASE=3Dconnexion Spacemouse Plus XT
+
+usb:v046DpC605*
+ ID_MODEL_FROM_DATABASE=3Dconnexion CADman
+
+usb:v046DpC606*
+ ID_MODEL_FROM_DATABASE=3Dconnexion Spacemouse Classic
+
+usb:v046DpC621*
+ ID_MODEL_FROM_DATABASE=3Dconnexion Spaceball 5000
+
 usb:v046DpC623*
  ID_MODEL_FROM_DATABASE=3Dconnexion Space Traveller 3D Mouse
 
@@ -7772,9 +7880,18 @@ usb:v046DpC626*
 usb:v046DpC627*
  ID_MODEL_FROM_DATABASE=3Dconnexion Space Explorer 3D Mouse
 
+usb:v046DpC628*
+ ID_MODEL_FROM_DATABASE=3Dconnexion Space Navigator for Notebooks
+
 usb:v046DpC629*
  ID_MODEL_FROM_DATABASE=3Dconnexion SpacePilot Pro 3D Mouse
 
+usb:v046DpC62B*
+ ID_MODEL_FROM_DATABASE=3Dconnexion Space Mouse Pro
+
+usb:v046DpC640*
+ ID_MODEL_FROM_DATABASE=NuLOOQ navigator
+
 usb:v046DpC702*
  ID_MODEL_FROM_DATABASE=Cordless Presenter
 
@@ -8609,6 +8726,9 @@ usb:v0480p0011*
 usb:v0480p0014*
  ID_MODEL_FROM_DATABASE=InTouch Module
 
+usb:v0480p0100*
+ ID_MODEL_FROM_DATABASE=Stor.E Slim USB 3.0
+
 usb:v0480pA006*
  ID_MODEL_FROM_DATABASE=External Disk 1.5TB
 
@@ -10043,6 +10163,9 @@ usb:v04A9p2225*
 usb:v04A9p2228*
  ID_MODEL_FROM_DATABASE=CanoScan 4400F
 
+usb:v04A9p2229*
+ ID_MODEL_FROM_DATABASE=CanoScan 8600F
+
 usb:v04A9p2602*
  ID_MODEL_FROM_DATABASE=MultiPASS C555
 
@@ -10988,6 +11111,9 @@ usb:v04A9p3233*
 usb:v04A9p3234*
  ID_MODEL_FROM_DATABASE=PowerShot SX150 IS
 
+usb:v04A9p3235*
+ ID_MODEL_FROM_DATABASE=PowerShot ELPH 510 HS / IXUS 1100 HS
+
 usb:v04A9p3236*
  ID_MODEL_FROM_DATABASE=PowerShot S100
 
@@ -11057,12 +11183,18 @@ usb:v04A9p325B*
 usb:v04A9p325C*
  ID_MODEL_FROM_DATABASE=PowerShot SX500 IS
 
+usb:v04A9p325E*
+ ID_MODEL_FROM_DATABASE=PowerShot N
+
 usb:v04A9p325F*
  ID_MODEL_FROM_DATABASE=PowerShot SX280 HS
 
 usb:v04A9p3260*
  ID_MODEL_FROM_DATABASE=PowerShot SX270 HS
 
+usb:v04A9p3261*
+ ID_MODEL_FROM_DATABASE=PowerShot A3500 IS
+
 usb:v04A9p3262*
  ID_MODEL_FROM_DATABASE=PowerShot A2600
 
@@ -11087,6 +11219,15 @@ usb:v04A9p3277*
 usb:v04A9p327D*
  ID_MODEL_FROM_DATABASE=Powershot ELPH 115 IS / IXUS 132
 
+usb:v04A9p3288*
+ ID_MODEL_FROM_DATABASE=Powershot ELPH 135 / IXUS 145
+
+usb:v04A9p3289*
+ ID_MODEL_FROM_DATABASE=PowerShot ELPH 340 HS / IXUS 265 HS
+
+usb:v04A9p328A*
+ ID_MODEL_FROM_DATABASE=PowerShot ELPH 150 IS / IXUS 155
+
 usb:v04AA*
  ID_VENDOR_FROM_DATABASE=DaeWoo Telecom, Ltd
 
@@ -11471,6 +11612,9 @@ usb:v04B4p2050*
 usb:v04B4p2830*
  ID_MODEL_FROM_DATABASE=Opera1 DVB-S (cold state)
 
+usb:v04B4p4235*
+ ID_MODEL_FROM_DATABASE=Monitor 02 Driver
+
 usb:v04B4p4381*
  ID_MODEL_FROM_DATABASE=SCAPS USC-1 Scanner Controller
 
@@ -12407,6 +12551,9 @@ usb:v04C5p10AE*
 usb:v04C5p10AF*
  ID_MODEL_FROM_DATABASE=fi-4220C2
 
+usb:v04C5p10C7*
+ ID_MODEL_FROM_DATABASE=fi-60f scanner
+
 usb:v04C5p10E0*
  ID_MODEL_FROM_DATABASE=fi-5120c Scanner
 
@@ -12981,7 +13128,7 @@ usb:v04D9p2519*
  ID_MODEL_FROM_DATABASE=Shenzhen LogoTech 2.4GHz receiver
 
 usb:v04D9p2832*
- ID_MODEL_FROM_DATABASE=1channel Telephone line recorder
+ ID_MODEL_FROM_DATABASE=HT82A832R Audio MCU
 
 usb:v04D9p2834*
  ID_MODEL_FROM_DATABASE=HT82A834R Audio MCU
@@ -13070,6 +13217,9 @@ usb:v04DAp2374*
 usb:v04DAp2451*
  ID_MODEL_FROM_DATABASE=HDC-SD9
 
+usb:v04DAp245B*
+ ID_MODEL_FROM_DATABASE=HC-X920K (3MOS Full HD video camcorder)
+
 usb:v04DAp2497*
  ID_MODEL_FROM_DATABASE=HDC-TM700
 
@@ -13535,6 +13685,9 @@ usb:v04E8p1006*
 usb:v04E8p130C*
  ID_MODEL_FROM_DATABASE=NX100
 
+usb:v04E8p1F05*
+ ID_MODEL_FROM_DATABASE=S2 Portable [JMicron] (500GB)
+
 usb:v04E8p1F06*
  ID_MODEL_FROM_DATABASE=HX-MU064DA portable harddisk
 
@@ -13670,6 +13823,9 @@ usb:v04E8p3310*
 usb:v04E8p3315*
  ID_MODEL_FROM_DATABASE=ML-2540 Series Laser Printer
 
+usb:v04E8p331E*
+ ID_MODEL_FROM_DATABASE=M262x/M282x Xpress Series Laser Printer
+
 usb:v04E8p3409*
  ID_MODEL_FROM_DATABASE=SCX-4216F Scanner
 
@@ -13718,6 +13874,9 @@ usb:v04E8p3420*
 usb:v04E8p3426*
  ID_MODEL_FROM_DATABASE=SCX-4500 Laser Printer
 
+usb:v04E8p342D*
+ ID_MODEL_FROM_DATABASE=SCX-4x28 Series
+
 usb:v04E8p344F*
  ID_MODEL_FROM_DATABASE=SCX-3400 Series
 
@@ -14058,7 +14217,7 @@ usb:v04E8p685B*
  ID_MODEL_FROM_DATABASE=GT-I9100 Phone [Galaxy S II] (mass storage mode)
 
 usb:v04E8p685C*
- ID_MODEL_FROM_DATABASE=GT-I9250 Phone [Galaxy Nexus]
+ ID_MODEL_FROM_DATABASE=GT-I9250 Phone [Galaxy Nexus] (Mass storage mode)
 
 usb:v04E8p685D*
  ID_MODEL_FROM_DATABASE=GT-I9100 Phone [Galaxy S II] (Download mode)
@@ -14067,7 +14226,10 @@ usb:v04E8p685E*
  ID_MODEL_FROM_DATABASE=GT-I9100 / GT-C3350 Phones (USB Debugging mode)
 
 usb:v04E8p6860*
- ID_MODEL_FROM_DATABASE=GT-I9100 Phone [Galaxy S II], GT-I9300 Phone [Galaxy S III], GT-P7500 [Galaxy Tab 10.1]
+ ID_MODEL_FROM_DATABASE=GT-I9100 Phone [Galaxy S II], GT-I9300 Phone [Galaxy S III], GT-P7500 [Galaxy Tab 10.1] , GT-I9500 [Galaxy S 4]
+
+usb:v04E8p6863*
+ ID_MODEL_FROM_DATABASE=GT-I9500 [Galaxy S4] / GT-I9250 [Galaxy Nexus] (network tethering)
 
 usb:v04E8p6865*
  ID_MODEL_FROM_DATABASE=GT-I9300 Phone [Galaxy S III] (PTP mode)
@@ -14075,6 +14237,9 @@ usb:v04E8p6865*
 usb:v04E8p6866*
  ID_MODEL_FROM_DATABASE=GT-I9300 Phone [Galaxy S III] (debugging mode)
 
+usb:v04E8p6868*
+ ID_MODEL_FROM_DATABASE=Escape Composite driver for Android Phones: Modem+Diagnostic+ADB
+
 usb:v04E8p6875*
  ID_MODEL_FROM_DATABASE=GT-B3710 Standalone LTE device (Commercial)
 
@@ -14342,6 +14507,12 @@ usb:v04F2pB057*
 usb:v04F2pB059*
  ID_MODEL_FROM_DATABASE=CKF7037 HP webcam
 
+usb:v04F2pB064*
+ ID_MODEL_FROM_DATABASE=CNA7137 Integrated Webcam
+
+usb:v04F2pB070*
+ ID_MODEL_FROM_DATABASE=Camera
+
 usb:v04F2pB071*
  ID_MODEL_FROM_DATABASE=2.0M UVC Webcam / CNF7129
 
@@ -14411,6 +14582,9 @@ usb:v04F2pB2B0*
 usb:v04F2pB2B9*
  ID_MODEL_FROM_DATABASE=Lenovo Integrated Camera UVC
 
+usb:v04F2pB2DA*
+ ID_MODEL_FROM_DATABASE=thinkpad t430s camera
+
 usb:v04F2pB2EA*
  ID_MODEL_FROM_DATABASE=Integrated Camera [ThinkPad]
 
@@ -14420,6 +14594,9 @@ usb:v04F2pB330*
 usb:v04F2pB354*
  ID_MODEL_FROM_DATABASE=UVC 1.00 device HD UVC WebCam
 
+usb:v04F2pB394*
+ ID_MODEL_FROM_DATABASE=Integrated Camera
+
 usb:v04F3*
  ID_VENDOR_FROM_DATABASE=Elan Microelectronics Corp.
 
@@ -15038,6 +15215,9 @@ usb:v04F9p01EB*
 usb:v04F9p01F4*
  ID_MODEL_FROM_DATABASE=MFC-5890CN
 
+usb:v04F9p0217*
+ ID_MODEL_FROM_DATABASE=MFC-8480DN
+
 usb:v04F9p0223*
  ID_MODEL_FROM_DATABASE=DCP-365CN
 
@@ -15158,9 +15338,15 @@ usb:v04FCp5331*
 usb:v04FCp5360*
  ID_MODEL_FROM_DATABASE=Sunplus Generic Digital Camera
 
+usb:v04FCp5563*
+ ID_MODEL_FROM_DATABASE=Digital Media Player MP3/WMA [The Sharper Image]
+
 usb:v04FCp5720*
  ID_MODEL_FROM_DATABASE=Card Reader Driver
 
+usb:v04FCp6333*
+ ID_MODEL_FROM_DATABASE=Siri A9 UVC chipset
+
 usb:v04FCp7333*
  ID_MODEL_FROM_DATABASE=Finet Technology Palmpix DC-85
 
@@ -15239,6 +15425,18 @@ usb:v0502p3325*
 usb:v0502p3341*
  ID_MODEL_FROM_DATABASE=Iconia tablet A500
 
+usb:v0502p33C3*
+ ID_MODEL_FROM_DATABASE=Liquid Gallant Duo E350
+
+usb:v0502p33C4*
+ ID_MODEL_FROM_DATABASE=Liquid Gallant Duo E350 (debug mode)
+
+usb:v0502p33C7*
+ ID_MODEL_FROM_DATABASE=Liquid Gallant Duo E350 (USB tethering)
+
+usb:v0502p33C8*
+ ID_MODEL_FROM_DATABASE=Liquid Gallant Duo E350 (debug mode, USB tethering)
+
 usb:v0502pD001*
  ID_MODEL_FROM_DATABASE=Divio NW801/DVC-V6+ Digital Camera
 
@@ -15446,6 +15644,9 @@ usb:v050Dp0416*
 usb:v050Dp0551*
  ID_MODEL_FROM_DATABASE=F6C550-AVR UPS
 
+usb:v050Dp065A*
+ ID_MODEL_FROM_DATABASE=F8T065BF Mini Bluetooth 4.0 Adapter
+
 usb:v050Dp0706*
  ID_MODEL_FROM_DATABASE=2-N-1 7-Port Hub (Lower half)
 
@@ -15497,6 +15698,9 @@ usb:v050Dp2103*
 usb:v050Dp21F1*
  ID_MODEL_FROM_DATABASE=N300 WLAN N Adapter [ISY]
 
+usb:v050Dp21F2*
+ ID_MODEL_FROM_DATABASE=RTL8192CU 802.11n WLAN Adapter [ISY IWL 4000]
+
 usb:v050Dp258A*
  ID_MODEL_FROM_DATABASE=F5U258 Host to Host cable
 
@@ -15728,6 +15932,9 @@ usb:v0525p1200*
 usb:v0525p1265*
  ID_MODEL_FROM_DATABASE=File-backed Storage Gadget
 
+usb:v0525p3424*
+ ID_MODEL_FROM_DATABASE=Lumidigm Venus fingerprint sensor
+
 usb:v0525pA0F0*
  ID_MODEL_FROM_DATABASE=Cambridge Electronic Devices Power1401 mk 2
 
@@ -16655,6 +16862,9 @@ usb:v054Cp035B*
 usb:v054Cp035C*
  ID_MODEL_FROM_DATABASE=NWZ-A726/A728/A729
 
+usb:v054Cp035F*
+ ID_MODEL_FROM_DATABASE=UP-DR200 Photo Printer
+
 usb:v054Cp0382*
  ID_MODEL_FROM_DATABASE=Memory Stick PRO-HG Duo Adaptor (MSAC-UAH1)
 
@@ -16694,9 +16904,18 @@ usb:v054Cp0485*
 usb:v054Cp04CB*
  ID_MODEL_FROM_DATABASE=WALKMAN NWZ-E354
 
+usb:v054Cp0541*
+ ID_MODEL_FROM_DATABASE=DSC-HX100V [Cybershot Digital Still Camera]
+
+usb:v054Cp0689*
+ ID_MODEL_FROM_DATABASE=Walkman NWZ-B173F
+
 usb:v054Cp06BB*
  ID_MODEL_FROM_DATABASE=WALKMAN NWZ-F805
 
+usb:v054Cp088C*
+ ID_MODEL_FROM_DATABASE=Portable Headphone Amplifier
+
 usb:v054Cp1000*
  ID_MODEL_FROM_DATABASE=Wireless Buzz! Receiver
 
@@ -16721,6 +16940,12 @@ usb:v0550p0004*
 usb:v0550p0005*
  ID_MODEL_FROM_DATABASE=InkJet Color Printer
 
+usb:v0550p000B*
+ ID_MODEL_FROM_DATABASE=Workcentre 24
+
+usb:v0550p014E*
+ ID_MODEL_FROM_DATABASE=CM215b Printer
+
 usb:v0551*
  ID_VENDOR_FROM_DATABASE=CompuTrend Systems, Inc.
 
@@ -17306,6 +17531,9 @@ usb:v056Ap0069*
 usb:v056Ap0081*
  ID_MODEL_FROM_DATABASE=Graphire Wireless 6x8
 
+usb:v056Ap0084*
+ ID_MODEL_FROM_DATABASE=Wireless adapter for Bamboo tablets
+
 usb:v056Ap0090*
  ID_MODEL_FROM_DATABASE=TPC90
 
@@ -17372,6 +17600,9 @@ usb:v056Ap00D1*
 usb:v056Ap00D3*
  ID_MODEL_FROM_DATABASE=Bamboo Fun (CTH-661)
 
+usb:v056Ap00D4*
+ ID_MODEL_FROM_DATABASE=Bamboo Pen (CTL-460)
+
 usb:v056Ap00D6*
  ID_MODEL_FROM_DATABASE=Bamboo Pen & Touch (CTH-460)
 
@@ -17390,6 +17621,12 @@ usb:v056Ap00F6*
 usb:v056Ap00F8*
  ID_MODEL_FROM_DATABASE=Cintiq 24HD touch (DTH-2400) tablet
 
+usb:v056Ap0307*
+ ID_MODEL_FROM_DATABASE=Cintiq Companion Hybrid 13HD (DTH-A1300) tablet
+
+usb:v056Ap0309*
+ ID_MODEL_FROM_DATABASE=Cintiq Companion Hybrid 13HD (DTH-A1300) touchscreen
+
 usb:v056Ap0400*
  ID_MODEL_FROM_DATABASE=PenPartner 4x5
 
@@ -18719,6 +18956,9 @@ usb:v0586p341E*
 usb:v0586p341F*
  ID_MODEL_FROM_DATABASE=NWD2205 802.11n Wireless N Adapter [Realtek RTL8192CU]
 
+usb:v0586p3425*
+ ID_MODEL_FROM_DATABASE=NWD6505 802.11a/b/g/n/ac Wireless Adapter [MediaTek MT7610U]
+
 usb:v0586p343E*
  ID_MODEL_FROM_DATABASE=N220 802.11bgn Wireless Adapter
 
@@ -18737,9 +18977,15 @@ usb:v058A*
 usb:v058B*
  ID_VENDOR_FROM_DATABASE=Infineon Technologies
 
+usb:v058Bp0015*
+ ID_MODEL_FROM_DATABASE=Flash Loader utility
+
 usb:v058Bp001C*
  ID_MODEL_FROM_DATABASE=Flash Drive
 
+usb:v058Bp0041*
+ ID_MODEL_FROM_DATABASE=Flash Loader utility
+
 usb:v058C*
  ID_VENDOR_FROM_DATABASE=In Focus Systems
 
@@ -18855,7 +19101,7 @@ usb:v058Fp6366*
  ID_MODEL_FROM_DATABASE=Multi Flash Reader
 
 usb:v058Fp6377*
- ID_MODEL_FROM_DATABASE=Multimedia Card Reader
+ ID_MODEL_FROM_DATABASE=AU6375 4-LUN card reader
 
 usb:v058Fp6386*
  ID_MODEL_FROM_DATABASE=Memory Card
@@ -18926,12 +19172,18 @@ usb:v058Fp9510*
 usb:v058Fp9520*
  ID_MODEL_FROM_DATABASE=EMV Certified Smart Card Reader
 
+usb:v058Fp9540*
+ ID_MODEL_FROM_DATABASE=AU9540 Smartcard Reader
+
 usb:v058Fp9720*
  ID_MODEL_FROM_DATABASE=USB-Serial Adapter
 
 usb:v058FpA014*
  ID_MODEL_FROM_DATABASE=Asus Integrated Webcam
 
+usb:v058FpB002*
+ ID_MODEL_FROM_DATABASE=Acer Integrated Webcam
+
 usb:v0590*
  ID_VENDOR_FROM_DATABASE=Omron Corp.
 
@@ -18983,6 +19235,9 @@ usb:v0596p0002*
 usb:v0596p0500*
  ID_MODEL_FROM_DATABASE=PCT Multitouch HID Controller
 
+usb:v0596p0543*
+ ID_MODEL_FROM_DATABASE=DELL XPS touchscreen
+
 usb:v0597*
  ID_VENDOR_FROM_DATABASE=Trisignal Communications
 
@@ -19055,6 +19310,9 @@ usb:v059Bp0061*
 usb:v059Bp006D*
  ID_MODEL_FROM_DATABASE=HipZip MP3 Player
 
+usb:v059Bp0070*
+ ID_MODEL_FROM_DATABASE=eGo Portable Hard Drive
+
 usb:v059Bp007C*
  ID_MODEL_FROM_DATABASE=Ultra Max USB/1394
 
@@ -19163,9 +19421,15 @@ usb:v059Fp0323*
 usb:v059Fp0421*
  ID_MODEL_FROM_DATABASE=Big Disk G465
 
+usb:v059Fp0525*
+ ID_MODEL_FROM_DATABASE=BigDisk Extreme 500
+
 usb:v059Fp0641*
  ID_MODEL_FROM_DATABASE=Mobile Hard Drive
 
+usb:v059Fp0829*
+ ID_MODEL_FROM_DATABASE=BigDisk Extreme+
+
 usb:v059Fp100C*
  ID_MODEL_FROM_DATABASE=Rugged Triple Interface Mobile Hard Drive
 
@@ -19190,6 +19454,9 @@ usb:v059Fp1049*
 usb:v059Fp1052*
  ID_MODEL_FROM_DATABASE=P'9220 Mobile Drive
 
+usb:v059Fp1064*
+ ID_MODEL_FROM_DATABASE=Rugged 16 and 32 GB
+
 usb:v059FpA601*
  ID_MODEL_FROM_DATABASE=HardDrive
 
@@ -19464,7 +19731,7 @@ usb:v05ACp0225*
  ID_MODEL_FROM_DATABASE=Internal Keyboard/Trackpad (JIS)
 
 usb:v05ACp0229*
- ID_MODEL_FROM_DATABASE=Internal Keyboard/Trackpad (MacBook Pro) (ANSI)
+ ID_MODEL_FROM_DATABASE=Internal Keyboard/Trackpad (ANSI)
 
 usb:v05ACp022A*
  ID_MODEL_FROM_DATABASE=Internal Keyboard/Trackpad (MacBook Pro) (ISO)
@@ -19545,7 +19812,7 @@ usb:v05ACp0302*
  ID_MODEL_FROM_DATABASE=Optical Mouse [Fujitsu]
 
 usb:v05ACp0304*
- ID_MODEL_FROM_DATABASE=Optical USB Mouse [Mitsumi]
+ ID_MODEL_FROM_DATABASE=Mighty Mouse [Mitsumi, M1152]
 
 usb:v05ACp0306*
  ID_MODEL_FROM_DATABASE=Optical USB Mouse [Fujitsu]
@@ -19577,6 +19844,9 @@ usb:v05ACp1003*
 usb:v05ACp1006*
  ID_MODEL_FROM_DATABASE=Hub in Aluminum Keyboard
 
+usb:v05ACp1008*
+ ID_MODEL_FROM_DATABASE=Mini DisplayPort to Dual-Link DVI Adapter
+
 usb:v05ACp1101*
  ID_MODEL_FROM_DATABASE=Speakers
 
@@ -19670,6 +19940,9 @@ usb:v05ACp1265*
 usb:v05ACp1266*
  ID_MODEL_FROM_DATABASE=iPod Nano 6.Gen
 
+usb:v05ACp1267*
+ ID_MODEL_FROM_DATABASE=iPod Nano 7.Gen
+
 usb:v05ACp1281*
  ID_MODEL_FROM_DATABASE=Apple Mobile Device [Recovery Mode]
 
@@ -19700,6 +19973,9 @@ usb:v05ACp1299*
 usb:v05ACp129A*
  ID_MODEL_FROM_DATABASE=iPad
 
+usb:v05ACp129C*
+ ID_MODEL_FROM_DATABASE=iPhone 4(CDMA)
+
 usb:v05ACp129E*
  ID_MODEL_FROM_DATABASE=iPod Touch 4.Gen
 
@@ -19712,15 +19988,30 @@ usb:v05ACp12A0*
 usb:v05ACp12A2*
  ID_MODEL_FROM_DATABASE=iPad 2 (3G; 64GB)
 
+usb:v05ACp12A3*
+ ID_MODEL_FROM_DATABASE=iPad 2 (CDMA)
+
+usb:v05ACp12A4*
+ ID_MODEL_FROM_DATABASE=iPad 3 (wifi)
+
+usb:v05ACp12A5*
+ ID_MODEL_FROM_DATABASE=iPad 3 (CDMA)
+
 usb:v05ACp12A6*
  ID_MODEL_FROM_DATABASE=iPad 3 (3G, 16 GB)
 
+usb:v05ACp12A8*
+ ID_MODEL_FROM_DATABASE=iPhone5/5C/5S
+
 usb:v05ACp12A9*
  ID_MODEL_FROM_DATABASE=iPad 2
 
 usb:v05ACp12AA*
  ID_MODEL_FROM_DATABASE=iPod Touch 5.Gen [A1421]
 
+usb:v05ACp12AB*
+ ID_MODEL_FROM_DATABASE=iPad 4 (WiFi, 32GB)
+
 usb:v05ACp1300*
  ID_MODEL_FROM_DATABASE=iPod Shuffle
 
@@ -19742,6 +20033,15 @@ usb:v05ACp1402*
 usb:v05ACp1500*
  ID_MODEL_FROM_DATABASE=SuperDrive [A1379]
 
+usb:v05ACp8005*
+ ID_MODEL_FROM_DATABASE=OHCI Root Hub Simulation
+
+usb:v05ACp8006*
+ ID_MODEL_FROM_DATABASE=EHCI Root Hub Simulation
+
+usb:v05ACp8007*
+ ID_MODEL_FROM_DATABASE=XHCI Root Hub USB 2.0 Simulation
+
 usb:v05ACp8202*
  ID_MODEL_FROM_DATABASE=HCF V.90 Data/Fax Modem
 
@@ -19838,6 +20138,9 @@ usb:v05ACp8510*
 usb:v05ACp911C*
  ID_MODEL_FROM_DATABASE=Hub in A1082 [Cinema HD Display 23"]
 
+usb:v05ACp9127*
+ ID_MODEL_FROM_DATABASE=Hub in Thunderbolt Display
+
 usb:v05ACp912F*
  ID_MODEL_FROM_DATABASE=Hub in 30" Cinema Display
 
@@ -20042,6 +20345,9 @@ usb:v05C6p9202*
 usb:v05C6p9203*
  ID_MODEL_FROM_DATABASE=Gobi Wireless Modem
 
+usb:v05C6p9205*
+ ID_MODEL_FROM_DATABASE=Gobi 2000
+
 usb:v05C6p9211*
  ID_MODEL_FROM_DATABASE=Acer Gobi Wireless Modem (QDL mode)
 
@@ -20123,9 +20429,18 @@ usb:v05C8p021A*
 usb:v05C8p0318*
  ID_MODEL_FROM_DATABASE=Webcam
 
+usb:v05C8p0361*
+ ID_MODEL_FROM_DATABASE=SunplusIT INC. HP Truevision HD Webcam
+
+usb:v05C8p036E*
+ ID_MODEL_FROM_DATABASE=Webcam
+
 usb:v05C8p0403*
  ID_MODEL_FROM_DATABASE=Webcam
 
+usb:v05C8p041B*
+ ID_MODEL_FROM_DATABASE=HP 2.0MP High Definition Webcam
+
 usb:v05C9*
  ID_VENDOR_FROM_DATABASE=Semtech Corp.
 
@@ -20861,6 +21176,9 @@ usb:v05DCp4D12*
 usb:v05DCp4D30*
  ID_MODEL_FROM_DATABASE=MP3 Player
 
+usb:v05DCpA209*
+ ID_MODEL_FROM_DATABASE=JumpDrive S70
+
 usb:v05DCpA300*
  ID_MODEL_FROM_DATABASE=JumpDrive2
 
@@ -20921,6 +21239,9 @@ usb:v05DCpA701*
 usb:v05DCpA731*
  ID_MODEL_FROM_DATABASE=JumpDrive FireFly
 
+usb:v05DCpA768*
+ ID_MODEL_FROM_DATABASE=JumpDrive Retrax
+
 usb:v05DCpA790*
  ID_MODEL_FROM_DATABASE=JumpDrive 2GB
 
@@ -20930,6 +21251,9 @@ usb:v05DCpA811*
 usb:v05DCpA813*
  ID_MODEL_FROM_DATABASE=16gB flash thumb drive
 
+usb:v05DCpA815*
+ ID_MODEL_FROM_DATABASE=JumpDrive V10
+
 usb:v05DCpB002*
  ID_MODEL_FROM_DATABASE=USB CF Reader
 
@@ -20939,6 +21263,9 @@ usb:v05DCpB018*
 usb:v05DCpB047*
  ID_MODEL_FROM_DATABASE=SDHC Reader [RW047-7000]
 
+usb:v05DCpBA02*
+ ID_MODEL_FROM_DATABASE=Workflow CFR1
+
 usb:v05DCpC753*
  ID_MODEL_FROM_DATABASE=JumpDrive TwistTurn
 
@@ -21084,7 +21411,7 @@ usb:v05E3p0607*
  ID_MODEL_FROM_DATABASE=Logitech G110 Hub
 
 usb:v05E3p0608*
- ID_MODEL_FROM_DATABASE=USB-2.0 4-Port HUB
+ ID_MODEL_FROM_DATABASE=Hub
 
 usb:v05E3p0610*
  ID_MODEL_FROM_DATABASE=4-port hub
@@ -21158,6 +21485,9 @@ usb:v05E3p0718*
 usb:v05E3p0719*
  ID_MODEL_FROM_DATABASE=SATA adapter
 
+usb:v05E3p0722*
+ ID_MODEL_FROM_DATABASE=SD/MMC card reader
+
 usb:v05E3p0723*
  ID_MODEL_FROM_DATABASE=GL827L SD/MMC/MS Flash Card Reader
 
@@ -21170,9 +21500,18 @@ usb:v05E3p0727*
 usb:v05E3p0731*
  ID_MODEL_FROM_DATABASE=GL3310 SATA 3Gb/s Bridge Controller
 
+usb:v05E3p0732*
+ ID_MODEL_FROM_DATABASE=All-in-One Cardreader
+
 usb:v05E3p0736*
  ID_MODEL_FROM_DATABASE=microSD Reader/Writer
 
+usb:v05E3p0741*
+ ID_MODEL_FROM_DATABASE=microSD Card Reader
+
+usb:v05E3p0743*
+ ID_MODEL_FROM_DATABASE=SDXC and microSDXC CardReader
+
 usb:v05E3p0760*
  ID_MODEL_FROM_DATABASE=USB 2.0 Card Reader/Writer
 
@@ -21296,6 +21635,9 @@ usb:v05F3p0081*
 usb:v05F3p00FF*
  ID_MODEL_FROM_DATABASE=VEC Footpedal
 
+usb:v05F3p0203*
+ ID_MODEL_FROM_DATABASE=Y-mouse Keyboard & Mouse Adapter
+
 usb:v05F3p020B*
  ID_MODEL_FROM_DATABASE=PS2 Adapter
 
@@ -21410,6 +21752,9 @@ usb:v05FEp0014*
 usb:v05FEp1010*
  ID_MODEL_FROM_DATABASE=Optical Wireless
 
+usb:v05FEp2001*
+ ID_MODEL_FROM_DATABASE=Microsoft Wireless Receiver 700
+
 usb:v05FF*
  ID_VENDOR_FROM_DATABASE=LeCroy Corp.
 
@@ -21434,6 +21779,9 @@ usb:v0603*
 usb:v0603p00F1*
  ID_MODEL_FROM_DATABASE=Keyboard
 
+usb:v0603p00F2*
+ ID_MODEL_FROM_DATABASE=Keyboard (Labtec Ultra Flat Keyboard)
+
 usb:v0603p6871*
  ID_MODEL_FROM_DATABASE=Mouse
 
@@ -21491,6 +21839,9 @@ usb:v060Bp2231*
 usb:v060Bp2270*
  ID_MODEL_FROM_DATABASE=Gigabyte K8100 Aivia Gaming Keyboard
 
+usb:v060Bp5253*
+ ID_MODEL_FROM_DATABASE=Thermaltake MEKA G-Unit Gaming Keyboard
+
 usb:v060Bp5811*
  ID_MODEL_FROM_DATABASE=ACK-571U Wireless Keyboard
 
@@ -21911,6 +22262,9 @@ usb:v064EpA110*
 usb:v064EpA114*
  ID_MODEL_FROM_DATABASE=Lemote Webcam
 
+usb:v064EpA116*
+ ID_MODEL_FROM_DATABASE=UVC 1.3MPixel WebCam
+
 usb:v064EpA136*
  ID_MODEL_FROM_DATABASE=Asus Integrated Webcam [CN031B]
 
@@ -22893,7 +23247,7 @@ usb:v068Ep00F4*
  ID_MODEL_FROM_DATABASE=Combatstick
 
 usb:v068Ep00FA*
- ID_MODEL_FROM_DATABASE=Flight Sim Pedals
+ ID_MODEL_FROM_DATABASE=Ch Throttle Quadrant
 
 usb:v068Ep00FF*
  ID_MODEL_FROM_DATABASE=Flight Sim Yoke
@@ -22955,6 +23309,9 @@ usb:v0698p9999*
 usb:v0699*
  ID_VENDOR_FROM_DATABASE=Tektronix, Inc.
 
+usb:v0699p0347*
+ ID_MODEL_FROM_DATABASE=AFG 3022B
+
 usb:v069A*
  ID_VENDOR_FROM_DATABASE=Askey Computer Corp.
 
@@ -23486,6 +23843,9 @@ usb:v06C2*
 usb:v06C2p0030*
  ID_MODEL_FROM_DATABASE=PhidgetRFID
 
+usb:v06C2p0031*
+ ID_MODEL_FROM_DATABASE=RFID reader
+
 usb:v06C2p0038*
  ID_MODEL_FROM_DATABASE=4-Motor PhidgetServo v3.0
 
@@ -23789,6 +24149,12 @@ usb:v06D3p0394*
 usb:v06D3p03A1*
  ID_MODEL_FROM_DATABASE=CP9550D/DW Port
 
+usb:v06D3p3B30*
+ ID_MODEL_FROM_DATABASE=CP-D70DW / CP-D707DW
+
+usb:v06D3p3B31*
+ ID_MODEL_FROM_DATABASE=CP-K60DW-S
+
 usb:v06D4*
  ID_VENDOR_FROM_DATABASE=Cisco Systems
 
@@ -24575,6 +24941,9 @@ usb:v072Fp1000*
 usb:v072Fp1001*
  ID_MODEL_FROM_DATABASE=PLDT Drive
 
+usb:v072Fp2200*
+ ID_MODEL_FROM_DATABASE=ACR122U
+
 usb:v072Fp8002*
  ID_MODEL_FROM_DATABASE=AET63 BioTRUSTKey
 
@@ -25046,6 +25415,12 @@ usb:v0765*
 usb:v0765p5001*
  ID_MODEL_FROM_DATABASE=Huey PRO Colorimeter
 
+usb:v0765p5020*
+ ID_MODEL_FROM_DATABASE=i1 Display Pro
+
+usb:v0765p6003*
+ ID_MODEL_FROM_DATABASE=ColorMunki Smile
+
 usb:v0765pD094*
  ID_MODEL_FROM_DATABASE=X-Rite DTP94 [Quato Silver Haze Pro]
 
@@ -25325,12 +25700,24 @@ usb:v0781p5530*
 usb:v0781p5567*
  ID_MODEL_FROM_DATABASE=Cruzer Blade
 
+usb:v0781p556C*
+ ID_MODEL_FROM_DATABASE=Ultra
+
+usb:v0781p556D*
+ ID_MODEL_FROM_DATABASE=Memory Vault
+
 usb:v0781p5571*
  ID_MODEL_FROM_DATABASE=Cruzer Fit
 
+usb:v0781p5576*
+ ID_MODEL_FROM_DATABASE=Cruzer Facet
+
 usb:v0781p5580*
  ID_MODEL_FROM_DATABASE=SDCZ80 Flash Drive
 
+usb:v0781p5581*
+ ID_MODEL_FROM_DATABASE=Ultra
+
 usb:v0781p5E10*
  ID_MODEL_FROM_DATABASE=Encrypted
 
@@ -25830,7 +26217,7 @@ usb:v07AAp001A*
  ID_MODEL_FROM_DATABASE=ULUSB-11 Key
 
 usb:v07AAp001C*
- ID_MODEL_FROM_DATABASE=CG-WLUSB2GTST 802.11g Wireless Adapter [Intersil ISL3887]
+ ID_MODEL_FROM_DATABASE=CG-WLUSB2GT 802.11g Wireless Adapter [Intersil ISL3880]
 
 usb:v07AAp002E*
  ID_MODEL_FROM_DATABASE=CG-WLUSB2GPX [Ralink RT2571W]
@@ -26114,6 +26501,9 @@ usb:v07B4p0114*
 usb:v07B4p0118*
  ID_MODEL_FROM_DATABASE=Mju Mini Digital/Mju Digital 500 Camera / Stylus 850 SW
 
+usb:v07B4p0125*
+ ID_MODEL_FROM_DATABASE=Tough TG-1 Camera
+
 usb:v07B4p0184*
  ID_MODEL_FROM_DATABASE=P-S100 port
 
@@ -26441,6 +26831,9 @@ usb:v07C4pA400*
 usb:v07C4pA600*
  ID_MODEL_FROM_DATABASE=Card Reader
 
+usb:v07C4pA604*
+ ID_MODEL_FROM_DATABASE=12-in-1 Card Reader
+
 usb:v07C4pAD01*
  ID_MODEL_FROM_DATABASE=Mass Storage Device
 
@@ -26474,6 +26867,9 @@ usb:v07C4pC010*
 usb:v07C5*
  ID_VENDOR_FROM_DATABASE=APG Cash Drawer
 
+usb:v07C5p0500*
+ ID_MODEL_FROM_DATABASE=Cash Drawer
+
 usb:v07C6*
  ID_VENDOR_FROM_DATABASE=ShareWave, Inc.
 
@@ -27140,6 +27536,9 @@ usb:v0803*
 usb:v0803p1300*
  ID_MODEL_FROM_DATABASE=V92 Faxmodem
 
+usb:v0803p3095*
+ ID_MODEL_FROM_DATABASE=V.92 56K Mini External Modem Model 3095
+
 usb:v0803p4310*
  ID_MODEL_FROM_DATABASE=4410a Wireless-G Adapter [Intersil ISL3887]
 
@@ -27713,6 +28112,9 @@ usb:v0846p9018*
 usb:v0846p9020*
  ID_MODEL_FROM_DATABASE=WNA3100(v1) Wireless-N 300 [Broadcom BCM43231]
 
+usb:v0846p9021*
+ ID_MODEL_FROM_DATABASE=WNA3100M(v1) Wireless-N 300 [Realtek RTL8192CU]
+
 usb:v0846p9030*
  ID_MODEL_FROM_DATABASE=WNA1100 Wireless-N 150 [Atheros AR9271]
 
@@ -27722,9 +28124,21 @@ usb:v0846p9040*
 usb:v0846p9041*
  ID_MODEL_FROM_DATABASE=WNA1000M 802.11bgn [Realtek RTL8188CUS]
 
+usb:v0846p9042*
+ ID_MODEL_FROM_DATABASE=On Networks N150MA 802.11bgn [Realtek RTL8188CUS]
+
+usb:v0846p9050*
+ ID_MODEL_FROM_DATABASE=A6200 802.11a/b/g/n/ac Wireless Adapter [Broadcom BCM43526]
+
+usb:v0846p9052*
+ ID_MODEL_FROM_DATABASE=A6100 AC600 DB Wireless Adapter [Realtek RTL8811AU]
+
 usb:v0846pA001*
  ID_MODEL_FROM_DATABASE=PA101 10 Mbps HPNA Home Phoneline RJ-1
 
+usb:v0846pF001*
+ ID_MODEL_FROM_DATABASE=On Networks N300MA 802.11bgn [Realtek RTL8192CU]
+
 usb:v084D*
  ID_VENDOR_FROM_DATABASE=Minton Optic Industry Co., Inc.
 
@@ -27875,8 +28289,17 @@ usb:v085Ap8027*
 usb:v085C*
  ID_VENDOR_FROM_DATABASE=ColorVision, Inc.
 
+usb:v085Cp0100*
+ ID_MODEL_FROM_DATABASE=Spyder 1
+
 usb:v085Cp0200*
- ID_MODEL_FROM_DATABASE=Monitor Spyder
+ ID_MODEL_FROM_DATABASE=Spyder 2
+
+usb:v085Cp0300*
+ ID_MODEL_FROM_DATABASE=Spyder 3
+
+usb:v085Cp0400*
+ ID_MODEL_FROM_DATABASE=Spyder 4
 
 usb:v0862*
  ID_VENDOR_FROM_DATABASE=Teletrol Systems, Inc.
@@ -28127,6 +28550,12 @@ usb:v0892*
 usb:v0892p0101*
  ID_MODEL_FROM_DATABASE=Smartdio Reader/Writer
 
+usb:v0894*
+ ID_VENDOR_FROM_DATABASE=TSI Incorporated
+
+usb:v0894p0010*
+ ID_MODEL_FROM_DATABASE=Remote NDIS Network Device
+
 usb:v0897*
  ID_VENDOR_FROM_DATABASE=Lauterbach
 
@@ -28181,6 +28610,18 @@ usb:v08A9p0015*
 usb:v08AE*
  ID_VENDOR_FROM_DATABASE=Macally (Mace Group, Inc.)
 
+usb:v08B0*
+ ID_VENDOR_FROM_DATABASE=Metrohm
+
+usb:v08B0p0006*
+ ID_MODEL_FROM_DATABASE=814 Sample Processor
+
+usb:v08B0p0015*
+ ID_MODEL_FROM_DATABASE=857 Titrando
+
+usb:v08B0p001A*
+ ID_MODEL_FROM_DATABASE=852 Titrando
+
 usb:v08B4*
  ID_VENDOR_FROM_DATABASE=Sorenson Vision, Inc.
 
@@ -29036,6 +29477,12 @@ usb:v0906*
 usb:v0908*
  ID_VENDOR_FROM_DATABASE=Siemens AG
 
+usb:v0908p01F4*
+ ID_MODEL_FROM_DATABASE=SIMATIC NET CP 5711
+
+usb:v0908p01FE*
+ ID_MODEL_FROM_DATABASE=SIMATIC NET PC Adapter A2
+
 usb:v0908p2701*
  ID_MODEL_FROM_DATABASE=ShenZhen SANZHAI Technology Co.,Ltd Spy Pen VGA
 
@@ -29285,12 +29732,18 @@ usb:v091Ep23CC*
 usb:v091Ep2459*
  ID_MODEL_FROM_DATABASE=GPSmap 62/78 series
 
+usb:v091Ep2491*
+ ID_MODEL_FROM_DATABASE=Edge 800
+
 usb:v091Ep2519*
  ID_MODEL_FROM_DATABASE=eTrex 30
 
 usb:v091Ep2535*
  ID_MODEL_FROM_DATABASE=Edge 800
 
+usb:v091Ep253C*
+ ID_MODEL_FROM_DATABASE=GPSmap 62sc
+
 usb:v091Ep255B*
  ID_MODEL_FROM_DATABASE=Nuvi 2505LM
 
@@ -29315,12 +29768,18 @@ usb:v0922p0007*
 usb:v0922p0009*
  ID_MODEL_FROM_DATABASE=LabelWriter 310
 
+usb:v0922p0019*
+ ID_MODEL_FROM_DATABASE=LabelWriter 400
+
 usb:v0922p001A*
  ID_MODEL_FROM_DATABASE=LabelWriter 400 Turbo
 
 usb:v0922p0020*
  ID_MODEL_FROM_DATABASE=LabelWriter 450
 
+usb:v0922p1001*
+ ID_MODEL_FROM_DATABASE=LabelManager PnP
+
 usb:v0923*
  ID_VENDOR_FROM_DATABASE=IC Media Corp.
 
@@ -29363,6 +29822,9 @@ usb:v0925*
 usb:v0925p0005*
  ID_MODEL_FROM_DATABASE=Gamtec.,Ltd SmartJoy PLUS Adapter
 
+usb:v0925p03E8*
+ ID_MODEL_FROM_DATABASE=Wii Classic Controller Adapter
+
 usb:v0925p3881*
  ID_MODEL_FROM_DATABASE=Saleae Logic
 
@@ -29417,6 +29879,9 @@ usb:v0930p000C*
 usb:v0930p0010*
  ID_MODEL_FROM_DATABASE=Gigabeat S (mtp)
 
+usb:v0930p0200*
+ ID_MODEL_FROM_DATABASE=Integrated Bluetooth (Taiyo Yuden)
+
 usb:v0930p0301*
  ID_MODEL_FROM_DATABASE=PCX1100U Cable Modem (WDM)
 
@@ -29483,6 +29948,9 @@ usb:v0930p070B*
 usb:v0930p0A07*
  ID_MODEL_FROM_DATABASE=WLM-10U1 802.11abgn Wireless Adapter [Ralink RT3572]
 
+usb:v0930p0A13*
+ ID_MODEL_FROM_DATABASE=AX88179 Gigabit Ethernet [Toshiba]
+
 usb:v0930p0B05*
  ID_MODEL_FROM_DATABASE=PX1220E-1G25 External hard drive
 
@@ -29691,7 +30159,7 @@ usb:v0939*
  ID_VENDOR_FROM_DATABASE=Lumberg, Inc.
 
 usb:v0939p0B15*
- ID_MODEL_FROM_DATABASE=Toshiba Stor.E Alu 2 1TB (PX1710E-1HJ0)
+ ID_MODEL_FROM_DATABASE=Toshiba Stor.E Alu 2
 
 usb:v093A*
  ID_VENDOR_FROM_DATABASE=Pixart Imaging, Inc.
@@ -29753,6 +30221,9 @@ usb:v093Ap260F*
 usb:v093Ap2621*
  ID_MODEL_FROM_DATABASE=PAC731x Trust Webcam
 
+usb:v093Ap2622*
+ ID_MODEL_FROM_DATABASE=Webcam Genius
+
 usb:v093Ap2624*
  ID_MODEL_FROM_DATABASE=Webcam
 
@@ -29951,6 +30422,9 @@ usb:v0951p1653*
 usb:v0951p1656*
  ID_MODEL_FROM_DATABASE=DataTraveler Ultimate G2
 
+usb:v0951p1665*
+ ID_MODEL_FROM_DATABASE=Digital DataTraveler SE9 64GB
+
 usb:v0951p1689*
  ID_MODEL_FROM_DATABASE=DataTraveler SE9
 
@@ -29993,6 +30467,9 @@ usb:v0957p0200*
 usb:v0957p0202*
  ID_MODEL_FROM_DATABASE=E-Video DC-350 Camera
 
+usb:v0957p0407*
+ ID_MODEL_FROM_DATABASE=33220A Waveform Generator
+
 usb:v0957p0518*
  ID_MODEL_FROM_DATABASE=82357B GPIB Interface
 
@@ -30033,7 +30510,7 @@ usb:v095Dp0001*
  ID_MODEL_FROM_DATABASE=Polycom ViaVideo
 
 usb:v0967*
- ID_VENDOR_FROM_DATABASE=Acer (??)
+ ID_VENDOR_FROM_DATABASE=Acer NeWeb Corp.
 
 usb:v0967p0204*
  ID_MODEL_FROM_DATABASE=WarpLink 802.11b Adapter
@@ -30056,6 +30533,12 @@ usb:v096Ep0807*
 usb:v0971*
  ID_VENDOR_FROM_DATABASE=Gretag-Macbeth AG
 
+usb:v0971p2000*
+ ID_MODEL_FROM_DATABASE=i1 Pro
+
+usb:v0971p2001*
+ ID_MODEL_FROM_DATABASE=i1 Monitor
+
 usb:v0971p2003*
  ID_MODEL_FROM_DATABASE=Eye-One display
 
@@ -30063,7 +30546,7 @@ usb:v0971p2005*
  ID_MODEL_FROM_DATABASE=Huey
 
 usb:v0971p2007*
- ID_MODEL_FROM_DATABASE=ColorMunki
+ ID_MODEL_FROM_DATABASE=ColorMunki Photo
 
 usb:v0973*
  ID_VENDOR_FROM_DATABASE=Schlumberger
@@ -30182,6 +30665,9 @@ usb:v099Ap0638*
 usb:v099Ap610C*
  ID_MODEL_FROM_DATABASE=EL-610 Super Mini Electron luminescent Keyboard
 
+usb:v099Ap713A*
+ ID_MODEL_FROM_DATABASE=WK-713 Multimedia Keyboard
+
 usb:v099Ap7160*
  ID_MODEL_FROM_DATABASE=Hyper Slim Keyboard
 
@@ -30375,7 +30861,10 @@ usb:v09D3*
  ID_VENDOR_FROM_DATABASE=Com One
 
 usb:v09D3p0001*
- ID_MODEL_FROM_DATABASE=ISDN TA
+ ID_MODEL_FROM_DATABASE=ISDN TA / Light Rider 128K
+
+usb:v09D3p000B*
+ ID_MODEL_FROM_DATABASE=Bluetooth Adapter class 1 [BlueLight]
 
 usb:v09D7*
  ID_VENDOR_FROM_DATABASE=Novatel Wireless
@@ -30596,6 +31085,12 @@ usb:v0A07p00DA*
 usb:v0A0B*
  ID_VENDOR_FROM_DATABASE=Cybex Computer Products Co.
 
+usb:v0A0D*
+ ID_VENDOR_FROM_DATABASE=Servergy, Inc
+
+usb:v0A0Dp2514*
+ ID_MODEL_FROM_DATABASE=CTS-1000 Internal Hub
+
 usb:v0A11*
  ID_VENDOR_FROM_DATABASE=Xentec, Inc.
 
@@ -30887,6 +31382,9 @@ usb:v0A48p5024*
 usb:v0A48p5025*
  ID_MODEL_FROM_DATABASE=Mass Storage Device
 
+usb:v0A4A*
+ ID_VENDOR_FROM_DATABASE=Ploytec GmbH
+
 usb:v0A4B*
  ID_VENDOR_FROM_DATABASE=Fujitsu Media Devices, Ltd
 
@@ -31217,6 +31715,9 @@ usb:v0A5CpBD11*
 usb:v0A5CpBD13*
  ID_MODEL_FROM_DATABASE=BCM4323 802.11abgn Wireless Adapter
 
+usb:v0A5CpBD16*
+ ID_MODEL_FROM_DATABASE=BCM4319 802.11bgn Wireless Adapter
+
 usb:v0A5CpBD17*
  ID_MODEL_FROM_DATABASE=BCM43236 802.11abgn Wireless Adapter
 
@@ -31775,6 +32276,9 @@ usb:v0AC8pC33F*
 usb:v0AC8pC429*
  ID_MODEL_FROM_DATABASE=Lenovo ThinkCentre Web Camera
 
+usb:v0AC8pC42D*
+ ID_MODEL_FROM_DATABASE=Lenovo IdeaCentre Web Camera
+
 usb:v0AC9*
  ID_VENDOR_FROM_DATABASE=Micro Solutions, Inc.
 
@@ -31823,6 +32327,9 @@ usb:v0ACDp0630*
 usb:v0ACDp0810*
  ID_MODEL_FROM_DATABASE=SecurePIN (IDPA-506100Y) PIN Pad
 
+usb:v0ACDp2030*
+ ID_MODEL_FROM_DATABASE=ValueMag Magnetic Stripe Reader
+
 usb:v0ACE*
  ID_VENDOR_FROM_DATABASE=ZyDAS
 
@@ -32024,6 +32531,12 @@ usb:v0AF9p0010*
 usb:v0AF9p0011*
  ID_MODEL_FROM_DATABASE=Micro Innovations IC50C Webcam
 
+usb:v0AFA*
+ ID_VENDOR_FROM_DATABASE=DMC Co., Ltd.
+
+usb:v0AFAp07D2*
+ ID_MODEL_FROM_DATABASE=Controller Board for Projected Capacitive Touch Screen DUS3000
+
 usb:v0AFC*
  ID_VENDOR_FROM_DATABASE=Zaptronix Ltd
 
@@ -32042,6 +32555,9 @@ usb:v0B00*
 usb:v0B05*
  ID_VENDOR_FROM_DATABASE=ASUSTek Computer, Inc.
 
+usb:v0B05p0001*
+ ID_MODEL_FROM_DATABASE=MeMO Pad HD 7 (CD-ROM mode)
+
 usb:v0B05p1101*
  ID_MODEL_FROM_DATABASE=Mass Storage (UISDMC4S)
 
@@ -32147,6 +32663,9 @@ usb:v0B05p1784*
 usb:v0B05p1786*
  ID_MODEL_FROM_DATABASE=USB-N10 802.11n Network Adapter [Realtek RTL8188SU]
 
+usb:v0B05p1788*
+ ID_MODEL_FROM_DATABASE=BT-270 Bluetooth Adapter
+
 usb:v0B05p1791*
  ID_MODEL_FROM_DATABASE=WL-167G v3 802.11n Adapter [Realtek RTL8188SU]
 
@@ -32159,12 +32678,18 @@ usb:v0B05p179E*
 usb:v0B05p179F*
  ID_MODEL_FROM_DATABASE=Eee Note EA800 (tablet mode)
 
+usb:v0B05p17A0*
+ ID_MODEL_FROM_DATABASE=Xonar U3 sound card
+
 usb:v0B05p17A1*
  ID_MODEL_FROM_DATABASE=Eee Note EA800 (mass storage mode)
 
 usb:v0B05p17AB*
  ID_MODEL_FROM_DATABASE=USB-N13 802.11n Network Adapter (rev. B1) [Realtek RTL8192CU]
 
+usb:v0B05p17C9*
+ ID_MODEL_FROM_DATABASE=USB-AC53 802.11a/b/g/n/ac Wireless Adapter [Broadcom BCM43526]
+
 usb:v0B05p4C80*
  ID_MODEL_FROM_DATABASE=Transformer Pad TF300TG
 
@@ -32183,6 +32708,12 @@ usb:v0B05p4D01*
 usb:v0B05p4DAF*
  ID_MODEL_FROM_DATABASE=Transformer Pad Infinity TF700 (Fastboot)
 
+usb:v0B05p5410*
+ ID_MODEL_FROM_DATABASE=MeMO Pad HD 7 (MTP mode)
+
+usb:v0B05p5412*
+ ID_MODEL_FROM_DATABASE=MeMO Pad HD 7 (PTP mode)
+
 usb:v0B05p6101*
  ID_MODEL_FROM_DATABASE=Cable Modem
 
@@ -32228,9 +32759,15 @@ usb:v0B0E*
 usb:v0B0Ep0420*
  ID_MODEL_FROM_DATABASE=Jabra SPEAK 510
 
+usb:v0B0Ep094D*
+ ID_MODEL_FROM_DATABASE=GN Netcom / Jabra REVO Wireless
+
 usb:v0B0Ep1022*
  ID_MODEL_FROM_DATABASE=Jabra PRO 9450, Type 9400BS (DECT Headset)
 
+usb:v0B0Ep2007*
+ ID_MODEL_FROM_DATABASE=GN 2000 Stereo Corded Headset
+
 usb:v0B0Ep620C*
  ID_MODEL_FROM_DATABASE=Jabra BT620s
 
@@ -32291,6 +32828,9 @@ usb:v0B33*
 usb:v0B33p0020*
  ID_MODEL_FROM_DATABASE=ShuttleXpress
 
+usb:v0B33p0700*
+ ID_MODEL_FROM_DATABASE=RollerMouse Pro
+
 usb:v0B37*
  ID_VENDOR_FROM_DATABASE=Hitachi ULSI Systems Co., Ltd
 
@@ -32687,6 +33227,9 @@ usb:v0B95p1720*
 usb:v0B95p1780*
  ID_MODEL_FROM_DATABASE=AX88178
 
+usb:v0B95p1790*
+ ID_MODEL_FROM_DATABASE=AX88179 Gigabit Ethernet
+
 usb:v0B95p7720*
  ID_MODEL_FROM_DATABASE=AX88772
 
@@ -32786,6 +33329,9 @@ usb:v0BAFp011B*
 usb:v0BAFp0121*
  ID_MODEL_FROM_DATABASE=USR5423 802.11bg Wireless Adapter [ZyDAS ZD1211B]
 
+usb:v0BAFp0303*
+ ID_MODEL_FROM_DATABASE=USR5637 56K Faxmodem
+
 usb:v0BAFp6112*
  ID_MODEL_FROM_DATABASE=FaxModem Model 5633
 
@@ -33332,6 +33878,9 @@ usb:v0BB4p0C01*
 usb:v0BB4p0C02*
  ID_MODEL_FROM_DATABASE=Dream / ADP1 / G1 / Magic / Tattoo (Debug)
 
+usb:v0BB4p0C03*
+ ID_MODEL_FROM_DATABASE=Android Phone [Fairphone First Edition (FP1)]
+
 usb:v0BB4p0C13*
  ID_MODEL_FROM_DATABASE=Diamond
 
@@ -33371,6 +33920,12 @@ usb:v0BB4p0CA2*
 usb:v0BB4p0CA5*
  ID_MODEL_FROM_DATABASE=Android Phone [Evo Shift 4G]
 
+usb:v0BB4p0CAE*
+ ID_MODEL_FROM_DATABASE=T-Mobile MyTouch 4G Slide [Doubleshot]
+
+usb:v0BB4p0DEA*
+ ID_MODEL_FROM_DATABASE=M7_UL [HTC One]
+
 usb:v0BB4p0FF8*
  ID_MODEL_FROM_DATABASE=Desire HD (Tethering Mode)
 
@@ -33443,18 +33998,33 @@ usb:v0BC2p2320*
 usb:v0BC2p3008*
  ID_MODEL_FROM_DATABASE=FreeAgent Desk 1TB
 
+usb:v0BC2p3101*
+ ID_MODEL_FROM_DATABASE=FreeAgent XTreme 640GB
+
+usb:v0BC2p3312*
+ ID_MODEL_FROM_DATABASE=SRD00F2 [Expansion Desktop Drive 2TB]
+
 usb:v0BC2p3320*
  ID_MODEL_FROM_DATABASE=SRD00F2 [Expansion Desktop Drive]
 
 usb:v0BC2p3332*
  ID_MODEL_FROM_DATABASE=Expansion
 
+usb:v0BC2p5020*
+ ID_MODEL_FROM_DATABASE=FreeAgent GoFlex
+
 usb:v0BC2p5021*
  ID_MODEL_FROM_DATABASE=FreeAgent GoFlex USB 2.0
 
+usb:v0BC2p5030*
+ ID_MODEL_FROM_DATABASE=FreeAgent GoFlex Upgrade Cable STAE104
+
 usb:v0BC2p5031*
  ID_MODEL_FROM_DATABASE=FreeAgent GoFlex USB 3.0
 
+usb:v0BC2p5070*
+ ID_MODEL_FROM_DATABASE=FreeAgent GoFlex Desk
+
 usb:v0BC2p50A1*
  ID_MODEL_FROM_DATABASE=FreeAgent GoFlex Desk
 
@@ -33476,6 +34046,9 @@ usb:v0BC2pA0A1*
 usb:v0BC2pA0A4*
  ID_MODEL_FROM_DATABASE=Backup Plus Desktop Drive
 
+usb:v0BC2pAB00*
+ ID_MODEL_FROM_DATABASE=Slim Portable Drive
+
 usb:v0BC3*
  ID_VENDOR_FROM_DATABASE=IPWireless, Inc.
 
@@ -33650,6 +34223,12 @@ usb:v0BDAp0184*
 usb:v0BDAp0186*
  ID_MODEL_FROM_DATABASE=Card Reader
 
+usb:v0BDAp0301*
+ ID_MODEL_FROM_DATABASE=multicard reader
+
+usb:v0BDAp1724*
+ ID_MODEL_FROM_DATABASE=RTL8723AU 802.11n WLAN Adapter
+
 usb:v0BDAp2831*
  ID_MODEL_FROM_DATABASE=RTL2831U DVB-T
 
@@ -33707,6 +34286,9 @@ usb:v0BDAp8198*
 usb:v0BDAp8199*
  ID_MODEL_FROM_DATABASE=RTL8187SU 802.11g WLAN Adapter
 
+usb:v0BDAp8812*
+ ID_MODEL_FROM_DATABASE=RTL8812AU 802.11a/b/g/n/ac WLAN Adapter
+
 usb:v0BDB*
  ID_VENDOR_FROM_DATABASE=Ericsson Business Mobile Networks BV
 
@@ -34038,7 +34620,7 @@ usb:v0C24p0019*
  ID_MODEL_FROM_DATABASE=Bluetooth Device
 
 usb:v0C24p0021*
- ID_MODEL_FROM_DATABASE=Bluetooth Device
+ ID_MODEL_FROM_DATABASE=Bluetooth Device (V2.1+EDR)
 
 usb:v0C24p0C24*
  ID_MODEL_FROM_DATABASE=Bluetooth Device(SAMPLE)
@@ -34142,6 +34724,9 @@ usb:v0C45*
 usb:v0C45p0011*
  ID_MODEL_FROM_DATABASE=EBUDDY
 
+usb:v0C45p0520*
+ ID_MODEL_FROM_DATABASE=MaxTrack Wireless Mouse
+
 usb:v0C45p1018*
  ID_MODEL_FROM_DATABASE=Compact Flash storage memory card reader
 
@@ -34496,9 +35081,15 @@ usb:v0C45p62C0*
 usb:v0C45p62E0*
  ID_MODEL_FROM_DATABASE=MSI Starcam Racer
 
+usb:v0C45p6300*
+ ID_MODEL_FROM_DATABASE=PC Microscope camera
+
 usb:v0C45p6310*
  ID_MODEL_FROM_DATABASE=Sonix USB 2.0 Camera
 
+usb:v0C45p6340*
+ ID_MODEL_FROM_DATABASE=Camera
+
 usb:v0C45p6341*
  ID_MODEL_FROM_DATABASE=Defender G-Lens 2577 HD720p Camera
 
@@ -34526,6 +35117,12 @@ usb:v0C45p6419*
 usb:v0C45p641D*
  ID_MODEL_FROM_DATABASE=1.3 MPixel Integrated Webcam
 
+usb:v0C45p643F*
+ ID_MODEL_FROM_DATABASE=Dell Integrated HD Webcam
+
+usb:v0C45p644D*
+ ID_MODEL_FROM_DATABASE=1.3 MPixel Integrated Webcam
+
 usb:v0C45p6480*
  ID_MODEL_FROM_DATABASE=Sonix 1.3 MP Laptop Integrated Webcam
 
@@ -34535,6 +35132,9 @@ usb:v0C45p648B*
 usb:v0C45p64BD*
  ID_MODEL_FROM_DATABASE=Sony Visual Communication Camera
 
+usb:v0C45p7401*
+ ID_MODEL_FROM_DATABASE=TEMPer Temperature Sensor
+
 usb:v0C45p7402*
  ID_MODEL_FROM_DATABASE=TEMPerHUM Temperature & Humidity Sensor
 
@@ -35390,6 +35990,9 @@ usb:v0CF3p3000*
 usb:v0CF3p3002*
  ID_MODEL_FROM_DATABASE=AR3011 Bluetooth
 
+usb:v0CF3p3004*
+ ID_MODEL_FROM_DATABASE=AR3012 Bluetooth 4.0
+
 usb:v0CF3p3005*
  ID_MODEL_FROM_DATABASE=AR3011 Bluetooth
 
@@ -35564,6 +36167,15 @@ usb:v0D35*
 usb:v0D3A*
  ID_VENDOR_FROM_DATABASE=Posiflex Technologies, Inc.
 
+usb:v0D3Ap0206*
+ ID_MODEL_FROM_DATABASE=Series 3xxx Cash Drawer
+
+usb:v0D3Ap0207*
+ ID_MODEL_FROM_DATABASE=Series 3xxx Cash Drawer
+
+usb:v0D3Ap0500*
+ ID_MODEL_FROM_DATABASE=Magnetic Stripe Reader
+
 usb:v0D3C*
  ID_VENDOR_FROM_DATABASE=Sri Cable Technology, Ltd
 
@@ -35573,6 +36185,9 @@ usb:v0D3D*
 usb:v0D3Dp0001*
  ID_MODEL_FROM_DATABASE=HID Keyboard
 
+usb:v0D3Dp0040*
+ ID_MODEL_FROM_DATABASE=PS/2 Adapter
+
 usb:v0D3E*
  ID_VENDOR_FROM_DATABASE=Fitcom, inc.
 
@@ -35651,6 +36266,9 @@ usb:v0D49p7010*
 usb:v0D49p7100*
  ID_MODEL_FROM_DATABASE=OneTouch II 300GB External Hard Disk
 
+usb:v0D49p7310*
+ ID_MODEL_FROM_DATABASE=OneTouch 4
+
 usb:v0D49p7410*
  ID_MODEL_FROM_DATABASE=Mobile Hard Disk Drive (1TB)
 
@@ -35858,6 +36476,9 @@ usb:v0D78*
 usb:v0D7A*
  ID_VENDOR_FROM_DATABASE=MARX Datentechnik GmbH
 
+usb:v0D7Ap0001*
+ ID_MODEL_FROM_DATABASE=CrypToken
+
 usb:v0D7B*
  ID_VENDOR_FROM_DATABASE=Wellco Technology Co., Ltd
 
@@ -36404,6 +37025,12 @@ usb:v0DB7*
 usb:v0DB7p0002*
  ID_MODEL_FROM_DATABASE=Goldpfeil P-LAN
 
+usb:v0DBA*
+ ID_VENDOR_FROM_DATABASE=Digidesign
+
+usb:v0DBAp3000*
+ ID_MODEL_FROM_DATABASE=Mbox 2
+
 usb:v0DBC*
  ID_VENDOR_FROM_DATABASE=A&D Medical
 
@@ -36776,6 +37403,9 @@ usb:v0DF6p0060*
 usb:v0DF6p0062*
  ID_MODEL_FROM_DATABASE=WLA-5000 802.11abgn [Ralink RT3572]
 
+usb:v0DF6p0072*
+ ID_MODEL_FROM_DATABASE=AX88179 Gigabit Ethernet [Sitecom]
+
 usb:v0DF6p061C*
  ID_MODEL_FROM_DATABASE=LN-028 Network USB 2.0 Adapter
 
@@ -37253,6 +37883,9 @@ usb:v0E79p14AD*
 usb:v0E79p150E*
  ID_MODEL_FROM_DATABASE=80 G9
 
+usb:v0E79p3001*
+ ID_MODEL_FROM_DATABASE=40 Titanium
+
 usb:v0E7B*
  ID_VENDOR_FROM_DATABASE=On-Tech Industry Co., Ltd
 
@@ -37292,9 +37925,15 @@ usb:v0E8Dp1806*
 usb:v0E8Dp1836*
  ID_MODEL_FROM_DATABASE=Samsung SE-S084 Super WriteMaster Slim External DVD writer
 
+usb:v0E8Dp2000*
+ ID_MODEL_FROM_DATABASE=MT65xx Preloader
+
 usb:v0E8Dp3329*
  ID_MODEL_FROM_DATABASE=Qstarz BT-Q1000XT
 
+usb:v0E8Dp763E*
+ ID_MODEL_FROM_DATABASE=MT7630e Bluetooth Adapter
+
 usb:v0E8F*
  ID_VENDOR_FROM_DATABASE=GreenAsia Inc.
 
@@ -37302,7 +37941,7 @@ usb:v0E8Fp0003*
  ID_MODEL_FROM_DATABASE=MaxFire Blaze2
 
 usb:v0E8Fp0012*
- ID_MODEL_FROM_DATABASE=USB Wireless 2.4GHz Gamepad
+ ID_MODEL_FROM_DATABASE=Joystick/Gamepad
 
 usb:v0E8Fp0016*
  ID_MODEL_FROM_DATABASE=4 port USB 1.1 hub UH-174
@@ -37436,9 +38075,18 @@ usb:v0EB7*
 usb:v0EB8*
  ID_VENDOR_FROM_DATABASE=Mettler Toledo
 
+usb:v0EB8p2200*
+ ID_MODEL_FROM_DATABASE=Ariva Scale
+
 usb:v0EB8pF000*
  ID_MODEL_FROM_DATABASE=PS60 Scale
 
+usb:v0EBB*
+ ID_VENDOR_FROM_DATABASE=Thermo Fisher Scientific
+
+usb:v0EBBp0002*
+ ID_MODEL_FROM_DATABASE=FT-IR Spectrometer
+
 usb:v0EBE*
  ID_VENDOR_FROM_DATABASE=VWeb Corp.
 
@@ -37559,6 +38207,12 @@ usb:v0EEFp0001*
 usb:v0EEFp0002*
  ID_MODEL_FROM_DATABASE=Touchscreen Controller(Professional)
 
+usb:v0EEFp7200*
+ ID_MODEL_FROM_DATABASE=Touchscreen Controller
+
+usb:v0EEFpA802*
+ ID_MODEL_FROM_DATABASE=eGalaxTouch EXC7920
+
 usb:v0EF0*
  ID_VENDOR_FROM_DATABASE=Hitachi Cable, Ltd
 
@@ -37670,6 +38324,12 @@ usb:v0F12*
 usb:v0F13*
  ID_VENDOR_FROM_DATABASE=Acetek Technology Co., Ltd
 
+usb:v0F14*
+ ID_VENDOR_FROM_DATABASE=Ingenico
+
+usb:v0F14p0012*
+ ID_MODEL_FROM_DATABASE=Vital'Act 3S
+
 usb:v0F18*
  ID_VENDOR_FROM_DATABASE=Finger Lakes Instrumentation
 
@@ -37754,6 +38414,12 @@ usb:v0F37*
 usb:v0F38*
  ID_VENDOR_FROM_DATABASE=Nien-Yi Industrial Corp.
 
+usb:v0F39*
+ ID_VENDOR_FROM_DATABASE=TG3 Electronics
+
+usb:v0F39p0876*
+ ID_MODEL_FROM_DATABASE=Keyboard [87 Francium Pro]
+
 usb:v0F3D*
  ID_VENDOR_FROM_DATABASE=Airprime, Incorporated
 
@@ -37841,6 +38507,9 @@ usb:v0F63*
 usb:v0F63p0010*
  ID_MODEL_FROM_DATABASE=Leapster Explorer
 
+usb:v0F63p0022*
+ ID_MODEL_FROM_DATABASE=Leap Reader
+
 usb:v0F63p0500*
  ID_MODEL_FROM_DATABASE=Fly Fusion
 
@@ -37928,6 +38597,9 @@ usb:v0F6Ep0404*
 usb:v0F73*
  ID_VENDOR_FROM_DATABASE=DFI
 
+usb:v0F78*
+ ID_VENDOR_FROM_DATABASE=Guntermann & Drunck GmbH
+
 usb:v0F7C*
  ID_VENDOR_FROM_DATABASE=DQ Technology, Inc.
 
@@ -38064,7 +38736,7 @@ usb:v0FCAp8001*
  ID_MODEL_FROM_DATABASE=Blackberry Handheld
 
 usb:v0FCAp8004*
- ID_MODEL_FROM_DATABASE=Blackberry Handheld
+ ID_MODEL_FROM_DATABASE=Blackberry
 
 usb:v0FCAp8007*
  ID_MODEL_FROM_DATABASE=Blackberry Handheld
@@ -38147,6 +38819,12 @@ usb:v0FCEp3138*
 usb:v0FCEp3149*
  ID_MODEL_FROM_DATABASE=Xperia X8
 
+usb:v0FCEp514F*
+ ID_MODEL_FROM_DATABASE=Xperia arc S [Adb-Enable Mode]
+
+usb:v0FCEp5169*
+ ID_MODEL_FROM_DATABASE=Xperia S [Adb-Enable Mode]
+
 usb:v0FCEp5177*
  ID_MODEL_FROM_DATABASE=Xperia Ion [Debug Mode]
 
@@ -38309,6 +38987,9 @@ usb:v0FCEpE166*
 usb:v0FCEpE167*
  ID_MODEL_FROM_DATABASE=XPERIA mini
 
+usb:v0FCEpF0FA*
+ ID_MODEL_FROM_DATABASE=Liveview micro display MN800 in DFU mode
+
 usb:v0FCF*
  ID_VENDOR_FROM_DATABASE=Dynastream Innovations, Inc.
 
@@ -38316,13 +38997,16 @@ usb:v0FCFp1003*
  ID_MODEL_FROM_DATABASE=ANT Development Board
 
 usb:v0FCFp1004*
- ID_MODEL_FROM_DATABASE=ANT2USB
+ ID_MODEL_FROM_DATABASE=ANTUSB Stick
 
 usb:v0FCFp1006*
  ID_MODEL_FROM_DATABASE=ANT Development Board
 
 usb:v0FCFp1008*
- ID_MODEL_FROM_DATABASE=Mini stick Suunto
+ ID_MODEL_FROM_DATABASE=ANTUSB2 Stick
+
+usb:v0FCFp1009*
+ ID_MODEL_FROM_DATABASE=ANTUSB-m Stick
 
 usb:v0FD0*
  ID_VENDOR_FROM_DATABASE=Tulip Computers B.V.
@@ -38378,6 +39062,12 @@ usb:v0FDAp0100*
 usb:v0FDC*
  ID_VENDOR_FROM_DATABASE=Micro Plus
 
+usb:v0FDE*
+ ID_VENDOR_FROM_DATABASE=Oregon Scientific
+
+usb:v0FDEpCA01*
+ ID_MODEL_FROM_DATABASE=WMRS200 weather station
+
 usb:v0FE0*
  ID_VENDOR_FROM_DATABASE=Osterhout Design Group
 
@@ -38471,6 +39161,9 @@ usb:v0FFF*
 usb:v1000*
  ID_VENDOR_FROM_DATABASE=Speed Tech Corp.
 
+usb:v1000p153B*
+ ID_MODEL_FROM_DATABASE=TerraTec Electronic GmbH
+
 usb:v1001*
  ID_VENDOR_FROM_DATABASE=Ritronics Components (S) Pte., Ltd
 
@@ -38504,6 +39197,9 @@ usb:v1004p618E*
 usb:v1004p618F*
  ID_MODEL_FROM_DATABASE=Ally/Optimus One
 
+usb:v1004p61C5*
+ ID_MODEL_FROM_DATABASE=P880 / Charge only
+
 usb:v1004p61C6*
  ID_MODEL_FROM_DATABASE=Vortex (msc)
 
@@ -38514,7 +39210,7 @@ usb:v1004p61F1*
  ID_MODEL_FROM_DATABASE=Optimus Android Phone [LG Software mode]
 
 usb:v1004p61F9*
- ID_MODEL_FROM_DATABASE=V909 G-Slate
+ ID_MODEL_FROM_DATABASE=Optimus (Various Models) MTP Mode
 
 usb:v1004p61FC*
  ID_MODEL_FROM_DATABASE=Optimus 3
@@ -38528,9 +39224,15 @@ usb:v1004p6300*
 usb:v1004p631C*
  ID_MODEL_FROM_DATABASE=Optimus Android Phone [MTP mode]
 
+usb:v1004p631D*
+ ID_MODEL_FROM_DATABASE=Optimus Android Phone (Camera/PTP Mode)
+
 usb:v1004p631E*
  ID_MODEL_FROM_DATABASE=Optimus Android Phone [Camera/PTP mode]
 
+usb:v1004p631F*
+ ID_MODEL_FROM_DATABASE=Optimus Android Phone (Charge Mode)
+
 usb:v1004p6356*
  ID_MODEL_FROM_DATABASE=Optimus Android Phone [Virtual CD mode]
 
@@ -38540,6 +39242,9 @@ usb:v1004p6800*
 usb:v1004p7000*
  ID_MODEL_FROM_DATABASE=LG LDP-7024D(LD)USB
 
+usb:v1004p91C8*
+ ID_MODEL_FROM_DATABASE=P880 / USB tethering
+
 usb:v1004pA400*
  ID_MODEL_FROM_DATABASE=Renoir (KC910)
 
@@ -38742,13 +39447,13 @@ usb:v1033p0068*
  ID_MODEL_FROM_DATABASE=3,5'' HDD case MD-231
 
 usb:v1038*
- ID_VENDOR_FROM_DATABASE=Ideazon, Inc.
+ ID_VENDOR_FROM_DATABASE=SteelSeries ApS
 
 usb:v1038p0100*
- ID_MODEL_FROM_DATABASE=Zboard
+ ID_MODEL_FROM_DATABASE=Ideazon Zboard
 
 usb:v1038p1361*
- ID_MODEL_FROM_DATABASE=Sensei
+ ID_MODEL_FROM_DATABASE=Ideazon Sensei
 
 usb:v1039*
  ID_VENDOR_FROM_DATABASE=devolo AG
@@ -38946,28 +39651,37 @@ usb:v1058p0704*
  ID_MODEL_FROM_DATABASE=Passport External HDD
 
 usb:v1058p070A*
- ID_MODEL_FROM_DATABASE=My Passport Essential SE
+ ID_MODEL_FROM_DATABASE=My Passport Essential SE, My Passport for Mac (WDBAAB)
 
 usb:v1058p071A*
  ID_MODEL_FROM_DATABASE=My Passport
 
 usb:v1058p0730*
- ID_MODEL_FROM_DATABASE=My Passport
+ ID_MODEL_FROM_DATABASE=My Passport Essential (WDBACY)
 
 usb:v1058p0740*
  ID_MODEL_FROM_DATABASE=My Passport
 
+usb:v1058p0741*
+ ID_MODEL_FROM_DATABASE=My Passport Ultra
+
 usb:v1058p0742*
  ID_MODEL_FROM_DATABASE=My Passport Essential SE
 
 usb:v1058p0748*
- ID_MODEL_FROM_DATABASE=My Passport 1TB USB 3.0
+ ID_MODEL_FROM_DATABASE=My Passport (WDBKXH)
+
+usb:v1058p0810*
+ ID_MODEL_FROM_DATABASE=My Passport Ultra
 
 usb:v1058p0900*
  ID_MODEL_FROM_DATABASE=MyBook Essential External HDD
 
 usb:v1058p0901*
- ID_MODEL_FROM_DATABASE=MyBook External HDD
+ ID_MODEL_FROM_DATABASE=My Book Essential Edition (Green Ring)
+
+usb:v1058p0902*
+ ID_MODEL_FROM_DATABASE=My Book Pro Edition
 
 usb:v1058p0903*
  ID_MODEL_FROM_DATABASE=My Book Premium Edition
@@ -38976,20 +39690,32 @@ usb:v1058p0910*
  ID_MODEL_FROM_DATABASE=MyBook Essential External HDD
 
 usb:v1058p1001*
- ID_MODEL_FROM_DATABASE=External Hard Disk [Elements]
+ ID_MODEL_FROM_DATABASE=Elements Desktop (WDE1U)
 
 usb:v1058p1003*
  ID_MODEL_FROM_DATABASE=Elements 1000 GB
 
 usb:v1058p1010*
- ID_MODEL_FROM_DATABASE=Elements External HDD
+ ID_MODEL_FROM_DATABASE=Elements Portable (WDBAAR)
 
 usb:v1058p1021*
- ID_MODEL_FROM_DATABASE=Elements 2TB
+ ID_MODEL_FROM_DATABASE=Elements Desktop (WDBAAU)
 
 usb:v1058p1023*
  ID_MODEL_FROM_DATABASE=Elements SE
 
+usb:v1058p1048*
+ ID_MODEL_FROM_DATABASE=Elements Portable (WDBU6Y)
+
+usb:v1058p10A2*
+ ID_MODEL_FROM_DATABASE=Elements SE Portable (WDBPCK)
+
+usb:v1058p10A8*
+ ID_MODEL_FROM_DATABASE=Elements Portable (WDBUZG)
+
+usb:v1058p1100*
+ ID_MODEL_FROM_DATABASE=My Book Essential Edition 2.0 (WDH1U)
+
 usb:v1058p1103*
  ID_MODEL_FROM_DATABASE=My Book Studio
 
@@ -39272,6 +39998,9 @@ usb:v106Fp0009*
 usb:v106Fp000A*
  ID_MODEL_FROM_DATABASE=CR10x Coin Recycler
 
+usb:v106Fp000C*
+ ID_MODEL_FROM_DATABASE=Xchange
+
 usb:v1076*
  ID_VENDOR_FROM_DATABASE=GCT Semiconductor, Inc.
 
@@ -39335,6 +40064,12 @@ usb:v1099*
 usb:v109A*
  ID_VENDOR_FROM_DATABASE=DATASOFT Systems GmbH
 
+usb:v109B*
+ ID_VENDOR_FROM_DATABASE=Hisense
+
+usb:v109Bp9118*
+ ID_MODEL_FROM_DATABASE=Medion P4013 Mobile
+
 usb:v109F*
  ID_VENDOR_FROM_DATABASE=eSOL Co., Ltd
 
@@ -39458,6 +40193,15 @@ usb:v10BF*
 usb:v10BFp0001*
  ID_MODEL_FROM_DATABASE=SmartHome PowerLinc
 
+usb:v10C3*
+ ID_VENDOR_FROM_DATABASE=Universal Laser Systems, Inc.
+
+usb:v10C3p00A4*
+ ID_MODEL_FROM_DATABASE=ULS PLS Series Laser Engraver Firmware Loader
+
+usb:v10C3p00A5*
+ ID_MODEL_FROM_DATABASE=ULS Print Support
+
 usb:v10C4*
  ID_VENDOR_FROM_DATABASE=Cygnal Integrated Products, Inc.
 
@@ -39509,6 +40253,12 @@ usb:v10C4p8461*
 usb:v10C4p8477*
  ID_MODEL_FROM_DATABASE=Balluff RFID Reader
 
+usb:v10C4p8496*
+ ID_MODEL_FROM_DATABASE=SiLabs Cypress FW downloader
+
+usb:v10C4p8497*
+ ID_MODEL_FROM_DATABASE=SiLabs Cypress EVB
+
 usb:v10C4p8605*
  ID_MODEL_FROM_DATABASE=dilitronics ESoLUX solar lighting controller
 
@@ -39527,6 +40277,9 @@ usb:v10C4p8863*
 usb:v10C4p8897*
  ID_MODEL_FROM_DATABASE=C8051F38x HDMI Splitter [UHBX]
 
+usb:v10C4p8918*
+ ID_MODEL_FROM_DATABASE=C8051F38x HDMI Audio Extractor [VSA-HA-DP]
+
 usb:v10C4pEA60*
  ID_MODEL_FROM_DATABASE=CP210x UART Bridge / myAVR mySmartUSB light
 
@@ -39908,6 +40661,9 @@ usb:v1141*
 usb:v1142*
  ID_VENDOR_FROM_DATABASE=CyberScan Technologies, Inc.
 
+usb:v1142p0709*
+ ID_MODEL_FROM_DATABASE=Cyberview High Speed Scanner
+
 usb:v1145*
  ID_VENDOR_FROM_DATABASE=Japan Radio Company
 
@@ -39938,6 +40694,9 @@ usb:v114D*
 usb:v114F*
  ID_VENDOR_FROM_DATABASE=Wavecom
 
+usb:v114Fp1234*
+ ID_MODEL_FROM_DATABASE=Fastrack Xtend FXT001 Modem
+
 usb:v115B*
  ID_VENDOR_FROM_DATABASE=Salix Technology Co., Ltd.
 
@@ -40128,7 +40887,7 @@ usb:v1199p683A*
  ID_MODEL_FROM_DATABASE=MC8785 Device
 
 usb:v1199p683C*
- ID_MODEL_FROM_DATABASE=MC8790 Device
+ ID_MODEL_FROM_DATABASE=Mobile Broadband 3G/UMTS (MC8790 Device)
 
 usb:v1199p6850*
  ID_MODEL_FROM_DATABASE=AirCard 880 Device
@@ -40196,6 +40955,12 @@ usb:v1199p9009*
 usb:v1199p900A*
  ID_MODEL_FROM_DATABASE=Gobi 2000 Wireless Modem
 
+usb:v1199p9055*
+ ID_MODEL_FROM_DATABASE=Gobi 9x15 Multimode 3G/4G LTE Modem (NAT mode)
+
+usb:v1199p9057*
+ ID_MODEL_FROM_DATABASE=Gobi 9x15 Multimode 3G/4G LTE Modem (IP passthrough mode)
+
 usb:v119A*
  ID_VENDOR_FROM_DATABASE=ZHAN QI Technology Co., Ltd
 
@@ -40238,12 +41003,30 @@ usb:v11ACp6565*
 usb:v11B0*
  ID_VENDOR_FROM_DATABASE=ATECH FLASH TECHNOLOGY
 
+usb:v11B0p6208*
+ ID_MODEL_FROM_DATABASE=PRO-28U
+
+usb:v11BE*
+ ID_VENDOR_FROM_DATABASE=R&D International NV
+
+usb:v11BEpF0A0*
+ ID_MODEL_FROM_DATABASE=Martin Maxxyz DMX
+
 usb:v11C5*
  ID_VENDOR_FROM_DATABASE=Inmax
 
 usb:v11C5p0521*
  ID_MODEL_FROM_DATABASE=IMT-0521 Smartcard Reader
 
+usb:v11CA*
+ ID_VENDOR_FROM_DATABASE=VeriFone Inc
+
+usb:v11CAp0207*
+ ID_MODEL_FROM_DATABASE=PIN Pad VX 810
+
+usb:v11CAp0220*
+ ID_MODEL_FROM_DATABASE=PIN Pad VX 805
+
 usb:v11DB*
  ID_VENDOR_FROM_DATABASE=Topfield Co., Ltd.
 
@@ -40274,6 +41057,9 @@ usb:v11F5p0005*
 usb:v11F5p0008*
  ID_MODEL_FROM_DATABASE=UMTS/HSDPA Data Card
 
+usb:v11F5p0101*
+ ID_MODEL_FROM_DATABASE=RCU Connect
+
 usb:v11F6*
  ID_VENDOR_FROM_DATABASE=Prolific
 
@@ -40284,7 +41070,7 @@ usb:v11F7*
  ID_VENDOR_FROM_DATABASE=Alcatel (?)
 
 usb:v11F7p02DF*
- ID_MODEL_FROM_DATABASE=TD10 Mobile phone USB cable
+ ID_MODEL_FROM_DATABASE=Serial cable (v2) for TD-10 Mobile Phone
 
 usb:v1203*
  ID_VENDOR_FROM_DATABASE=TSC Auto ID Technology Co., Ltd
@@ -40352,6 +41138,9 @@ usb:v1228p0012*
 usb:v1228p0015*
  ID_MODEL_FROM_DATABASE=TPaq21/MPaq21 Datalogger
 
+usb:v1228p584C*
+ ID_MODEL_FROM_DATABASE=XL2 Logger
+
 usb:v1230*
  ID_VENDOR_FROM_DATABASE=Chipidea-Microelectronica, S.A.
 
@@ -40377,20 +41166,110 @@ usb:v1235*
  ID_VENDOR_FROM_DATABASE=Novation EMS
 
 usb:v1235p0001*
- ID_MODEL_FROM_DATABASE=ReMOTE Audio/XStation
+ ID_MODEL_FROM_DATABASE=ReMOTE Audio/XStation First Edition
 
 usb:v1235p0002*
  ID_MODEL_FROM_DATABASE=Speedio
 
 usb:v1235p0003*
- ID_MODEL_FROM_DATABASE=ReMOTE ZeRO SL
+ ID_MODEL_FROM_DATABASE=RemoteSL + ZeroSL
+
+usb:v1235p0004*
+ ID_MODEL_FROM_DATABASE=ReMOTE LE
+
+usb:v1235p0005*
+ ID_MODEL_FROM_DATABASE=XIOSynth [First Edition]
+
+usb:v1235p0006*
+ ID_MODEL_FROM_DATABASE=XStation
+
+usb:v1235p0007*
+ ID_MODEL_FROM_DATABASE=XIOSynth
+
+usb:v1235p0008*
+ ID_MODEL_FROM_DATABASE=ReMOTE SL Compact
+
+usb:v1235p0009*
+ ID_MODEL_FROM_DATABASE=nIO
+
+usb:v1235p000A*
+ ID_MODEL_FROM_DATABASE=Nocturn
+
+usb:v1235p000B*
+ ID_MODEL_FROM_DATABASE=ReMOTE SL MkII
+
+usb:v1235p000C*
+ ID_MODEL_FROM_DATABASE=ZeRO MkII
+
+usb:v1235p000E*
+ ID_MODEL_FROM_DATABASE=Launchpad
+
+usb:v1235p0010*
+ ID_MODEL_FROM_DATABASE=Saffire 6
+
+usb:v1235p0011*
+ ID_MODEL_FROM_DATABASE=Ultranova
+
+usb:v1235p0012*
+ ID_MODEL_FROM_DATABASE=Nocturn Keyboard
+
+usb:v1235p0013*
+ ID_MODEL_FROM_DATABASE=VRM Box
+
+usb:v1235p0014*
+ ID_MODEL_FROM_DATABASE=VRM Box Audio Class (2-out)
+
+usb:v1235p0015*
+ ID_MODEL_FROM_DATABASE=Dicer
+
+usb:v1235p0016*
+ ID_MODEL_FROM_DATABASE=Ultranova
+
+usb:v1235p0018*
+ ID_MODEL_FROM_DATABASE=Twitch
+
+usb:v1235p0019*
+ ID_MODEL_FROM_DATABASE=Impulse 25
+
+usb:v1235p001A*
+ ID_MODEL_FROM_DATABASE=Impulse 49
+
+usb:v1235p001B*
+ ID_MODEL_FROM_DATABASE=Impulse 61
 
 usb:v1235p4661*
  ID_MODEL_FROM_DATABASE=ReMOTE25
 
+usb:v1235p8000*
+ ID_MODEL_FROM_DATABASE=Scarlett 18i6
+
+usb:v1235p8002*
+ ID_MODEL_FROM_DATABASE=Scarlett 8i6
+
 usb:v1235p8006*
  ID_MODEL_FROM_DATABASE=Focusrite Scarlett 2i2
 
+usb:v1235p8008*
+ ID_MODEL_FROM_DATABASE=Saffire 6
+
+usb:v1235p800A*
+ ID_MODEL_FROM_DATABASE=Scarlett 2i4
+
+usb:v1235p800C*
+ ID_MODEL_FROM_DATABASE=Scarlett 18i20
+
+usb:v1235p800E*
+ ID_MODEL_FROM_DATABASE=iTrack Solo
+
+usb:v1235p8010*
+ ID_MODEL_FROM_DATABASE=Forte
+
+usb:v1235p8012*
+ ID_MODEL_FROM_DATABASE=Scarlett 6i6
+
+usb:v1235p8014*
+ ID_MODEL_FROM_DATABASE=Scarlett 18i8
+
 usb:v1241*
  ID_VENDOR_FROM_DATABASE=Belkin
 
@@ -40410,7 +41289,7 @@ usb:v1241p1166*
  ID_MODEL_FROM_DATABASE=MI-2150 Trust Mouse
 
 usb:v1241p1177*
- ID_MODEL_FROM_DATABASE=F8E842-DL Mouse
+ ID_MODEL_FROM_DATABASE=Mouse [HT82M21A]
 
 usb:v1241p1503*
  ID_MODEL_FROM_DATABASE=Keyboard
@@ -40463,6 +41342,9 @@ usb:v125Fp312A*
 usb:v125Fp312B*
  ID_MODEL_FROM_DATABASE=Superior S102 Pro
 
+usb:v125FpA22A*
+ ID_MODEL_FROM_DATABASE=DashDrive Elite HE720 500GB
+
 usb:v125FpA91A*
  ID_MODEL_FROM_DATABASE=Portable HDD CH91
 
@@ -40478,6 +41360,9 @@ usb:v125FpC93A*
 usb:v125FpC96A*
  ID_MODEL_FROM_DATABASE=C906 Flash Drive
 
+usb:v125FpCB10*
+ ID_MODEL_FROM_DATABASE=Dash Drive UV100
+
 usb:v1260*
  ID_VENDOR_FROM_DATABASE=Standard Microsystems Corp.
 
@@ -40637,6 +41522,9 @@ usb:v1283p0150*
 usb:v1286*
  ID_VENDOR_FROM_DATABASE=Marvell Semiconductor, Inc.
 
+usb:v1286p00BC*
+ ID_MODEL_FROM_DATABASE=Marvell JTAG Probe
+
 usb:v1286p1FAB*
  ID_MODEL_FROM_DATABASE=88W8338 [Libertas] 802.11g
 
@@ -40733,6 +41621,12 @@ usb:v12C4p0006*
 usb:v12C4p0008*
  ID_MODEL_FROM_DATABASE=Teleprompter Foot Control (v1)
 
+usb:v12CF*
+ ID_VENDOR_FROM_DATABASE=DEXIN
+
+usb:v12CFp0170*
+ ID_MODEL_FROM_DATABASE=Tt eSPORTS BLACK Gaming mouse
+
 usb:v12D1*
  ID_VENDOR_FROM_DATABASE=Huawei Technologies Co., Ltd.
 
@@ -40802,9 +41696,15 @@ usb:v12D1p14CF*
 usb:v12D1p14D1*
  ID_MODEL_FROM_DATABASE=K3770 3G Modem (Mass Storage Mode)
 
+usb:v12D1p14DB*
+ ID_MODEL_FROM_DATABASE=E353/E3131
+
 usb:v12D1p14F1*
  ID_MODEL_FROM_DATABASE=Gobi 3000 HSPA+ Modem
 
+usb:v12D1p14FE*
+ ID_MODEL_FROM_DATABASE=Modem (Mass Storage Mode)
+
 usb:v12D1p1501*
  ID_MODEL_FROM_DATABASE=Pulse
 
@@ -40826,6 +41726,9 @@ usb:v12D1p1521*
 usb:v12D1p155A*
  ID_MODEL_FROM_DATABASE=R205 Mobile WiFi (CD-ROM mode)
 
+usb:v12D1p1575*
+ ID_MODEL_FROM_DATABASE=K5150 LTE modem
+
 usb:v12D1p1805*
  ID_MODEL_FROM_DATABASE=AT&T Go Phone U2800A phone
 
@@ -41376,7 +42279,7 @@ usb:v13BA*
  ID_VENDOR_FROM_DATABASE=PCPlay
 
 usb:v13BAp0001*
- ID_MODEL_FROM_DATABASE=König Electronic CMP-KEYPAD12 Numeric Keypad
+ ID_MODEL_FROM_DATABASE=Konig Electronic CMP-KEYPAD12 Numeric Keypad
 
 usb:v13BAp0017*
  ID_MODEL_FROM_DATABASE=PS/2 Keyboard+Mouse Adapter
@@ -44096,6 +44999,9 @@ usb:v1740p9801*
 usb:v1743*
  ID_VENDOR_FROM_DATABASE=General Atomics
 
+usb:v1748*
+ ID_VENDOR_FROM_DATABASE=MQP Electronics
+
 usb:v174C*
  ID_VENDOR_FROM_DATABASE=ASMedia Technology Inc.
 
@@ -46569,7 +47475,7 @@ usb:v1C7B*
  ID_VENDOR_FROM_DATABASE=LUXSHARE PRECISION INDUSTRY (SHENZHEN) CO., LTD.
 
 usb:v1C83*
- ID_VENDOR_FROM_DATABASE=Schomäcker GmbH
+ ID_VENDOR_FROM_DATABASE=Schomaecker GmbH
 
 usb:v1C83p0001*
  ID_MODEL_FROM_DATABASE=RS150 V2
@@ -47678,6 +48584,12 @@ usb:v2047*
 usb:v2047p0200*
  ID_MODEL_FROM_DATABASE=MSP430 USB HID Bootstrap Loader
 
+usb:v2047p0855*
+ ID_MODEL_FROM_DATABASE=Invensense Embedded MotionApp HID Sensor
+
+usb:v2047p0964*
+ ID_MODEL_FROM_DATABASE=Inventio Software MSP430
+
 usb:v2080*
  ID_VENDOR_FROM_DATABASE=Barnes & Noble
 
